<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Makes plots for every region while summarizing the annotation — plotRegionCoverage • derfinderPlot</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Makes plots for every region while summarizing the annotation — plotRegionCoverage" />
<meta property="og:description" content="This function takes the regions found in calculatePvalues
and assigns them genomic states contructed with
makeGenomicState. The main workhorse functions are
countOverlaps and findOverlaps. For an
alternative plot check plotCluster which is much slower and we
recommend it's use only after quickly checking the results with this
function." />






<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  

  

  </head>

  <body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">derfinderPlot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.27.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/derfinderPlot.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/leekgroup/derfinderPlot/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Makes plots for every region while summarizing the annotation</h1>
    <small class="dont-index">Source: <a href='https://github.com/leekgroup/derfinderPlot/blob/master/R/plotRegionCoverage.R'><code>R/plotRegionCoverage.R</code></a></small>
    <div class="hidden name"><code>plotRegionCoverage.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function takes the regions found in <a href='https://rdrr.io/pkg/derfinder/man/calculatePvalues.html'>calculatePvalues</a>
and assigns them genomic states contructed with
<a href='https://rdrr.io/pkg/derfinder/man/makeGenomicState.html'>makeGenomicState</a>. The main workhorse functions are
<a href='https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html'>countOverlaps</a> and <a href='https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html'>findOverlaps</a>. For an
alternative plot check <a href='plotCluster.html'>plotCluster</a> which is much slower and we
recommend it's use only after quickly checking the results with this
function.</p>
    </div>

    <div class="ref-usage sourceCode"><pre class='sourceCode r'><code><span class='fu'>plotRegionCoverage</span><span class='op'>(</span>
  <span class='va'>regions</span>,
  <span class='va'>regionCoverage</span>,
  <span class='va'>groupInfo</span>,
  <span class='va'>nearestAnnotation</span>,
  <span class='va'>annotatedRegions</span>,
  txdb <span class='op'>=</span> <span class='cn'>NULL</span>,
  whichRegions <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq_len</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>min</a></span><span class='op'>(</span><span class='fl'>100</span>, <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='va'>regions</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>,
  colors <span class='op'>=</span> <span class='cn'>NULL</span>,
  scalefac <span class='op'>=</span> <span class='fl'>32</span>,
  ask <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/interactive.html'>interactive</a></span><span class='op'>(</span><span class='op'>)</span>,
  ylab <span class='op'>=</span> <span class='st'>"Coverage"</span>,
  verbose <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></code></pre></div>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>regions</th>
      <td><p>The <code>$regions</code> output from
<a href='https://rdrr.io/pkg/derfinder/man/calculatePvalues.html'>calculatePvalues</a>.</p></td>
    </tr>
    <tr>
      <th>regionCoverage</th>
      <td><p>The output from <a href='https://rdrr.io/pkg/derfinder/man/getRegionCoverage.html'>getRegionCoverage</a>
used on <code>regions</code>.</p></td>
    </tr>
    <tr>
      <th>groupInfo</th>
      <td><p>A factor specifying the group membership of each sample. It
will be used to color the samples by group.</p></td>
    </tr>
    <tr>
      <th>nearestAnnotation</th>
      <td><p>The output from <a href='https://rdrr.io/pkg/bumphunter/man/matchGenes.html'>matchGenes</a>
used on <code>regions</code>.</p></td>
    </tr>
    <tr>
      <th>annotatedRegions</th>
      <td><p>The output from <a href='https://rdrr.io/pkg/derfinder/man/annotateRegions.html'>annotateRegions</a>
used on <code>regions</code>.</p></td>
    </tr>
    <tr>
      <th>txdb</th>
      <td><p>A <a href='https://rdrr.io/pkg/GenomicFeatures/man/TxDb-class.html'>TxDb</a> object. If specified, transcript
annotation will be extracted from this object and used to plot the
transcripts.</p></td>
    </tr>
    <tr>
      <th>whichRegions</th>
      <td><p>An integer vector with the index of the regions to plot.</p></td>
    </tr>
    <tr>
      <th>colors</th>
      <td><p>If <code>NULL</code> then <a href='https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html'>brewer.pal</a> with the
<code>'Dark2'</code> color scheme is used.</p></td>
    </tr>
    <tr>
      <th>scalefac</th>
      <td><p>The parameter used in <a href='https://rdrr.io/pkg/derfinder/man/preprocessCoverage.html'>preprocessCoverage</a>.</p></td>
    </tr>
    <tr>
      <th>ask</th>
      <td><p>If <code>TRUE</code> then the user is prompted before each plot is made.</p></td>
    </tr>
    <tr>
      <th>ylab</th>
      <td><p>The name of the of the Y axis.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>If <code>TRUE</code> basic status updates will be printed along the
way.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A plot for every region showing the coverage of each sample at each
base of the region as well as the summarized annotation information.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><a href='https://rdrr.io/pkg/derfinder/man/calculatePvalues.html'>calculatePvalues</a>,
<a href='https://rdrr.io/pkg/derfinder/man/getRegionCoverage.html'>getRegionCoverage</a>,
<a href='https://rdrr.io/pkg/bumphunter/man/matchGenes.html'>matchGenes</a>, <a href='https://rdrr.io/pkg/derfinder/man/annotateRegions.html'>annotateRegions</a>,
<a href='plotCluster.html'>plotCluster</a></p></div>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Andrew Jaffe, Leonardo Collado-Torres</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <div class="ref-examples sourceCode"><pre class='sourceCode r'><code><span class='r-in'><span class='co'>## Load data</span></span>
<span class='r-in'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='st'><a href='https://github.com/lcolladotor/derfinder'>"derfinder"</a></span><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## Annotate regions, first two regions only</span></span>
<span class='r-in'><span class='va'>regions</span> <span class='op'>&lt;-</span> <span class='va'>genomeRegions</span><span class='op'>$</span><span class='va'>regions</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>2</span><span class='op'>]</span></span>
<span class='r-in'><span class='va'>annotatedRegions</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/derfinder/man/annotateRegions.html'>annotateRegions</a></span><span class='op'>(</span></span>
<span class='r-in'>    regions <span class='op'>=</span> <span class='va'>regions</span>,</span>
<span class='r-in'>    genomicState <span class='op'>=</span> <span class='va'>genomicState</span><span class='op'>$</span><span class='va'>fullGenome</span>, minoverlap <span class='op'>=</span> <span class='fl'>1</span></span>
<span class='r-in'><span class='op'>)</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-08-10 01:22:35 annotateRegions: counting</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-08-10 01:22:36 annotateRegions: annotating</span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## Find nearest annotation with bumphunter::matchGenes()</span></span>
<span class='r-in'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='st'><a href='https://github.com/rafalab/bumphunter'>"bumphunter"</a></span><span class='op'>)</span></span>
<span class='r-in'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='st'>"TxDb.Hsapiens.UCSC.hg19.knownGene"</span><span class='op'>)</span></span>
<span class='r-in'><span class='va'>genes</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/bumphunter/man/annotateTranscripts.html'>annotateTranscripts</a></span><span class='op'>(</span>txdb <span class='op'>=</span> <span class='va'>TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class='op'>)</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> No annotationPackage supplied. Trying org.Hs.eg.db.</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Getting TSS and TSE.</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Getting CSS and CSE.</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Getting exons.</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> Annotating genes.</span>
<span class='r-in'><span class='va'>nearestAnnotation</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/bumphunter/man/matchGenes.html'>matchGenes</a></span><span class='op'>(</span>x <span class='op'>=</span> <span class='va'>regions</span>, subject <span class='op'>=</span> <span class='va'>genes</span><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## Obtain fullCov object</span></span>
<span class='r-in'><span class='va'>fullCov</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span><span class='st'>"21"</span> <span class='op'>=</span> <span class='va'>genomeDataRaw</span><span class='op'>$</span><span class='va'>coverage</span><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## Assign chr lengths using hg19 information</span></span>
<span class='r-in'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='st'><a href='https://bioconductor.org/packages/GenomicRanges'>"GenomicRanges"</a></span><span class='op'>)</span></span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html'>seqlengths</a></span><span class='op'>(</span><span class='va'>regions</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html'>seqlengths</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/GenomeInfoDb/man/getChromInfoFromUCSC.html'>getChromInfoFromUCSC</a></span><span class='op'>(</span><span class='st'>"hg19"</span>,</span>
<span class='r-in'>    as.Seqinfo <span class='op'>=</span> <span class='cn'>TRUE</span></span>
<span class='r-in'><span class='op'>)</span><span class='op'>)</span><span class='op'>[</span></span>
<span class='r-in'>    <span class='fu'><a href='https://rdrr.io/pkg/GenomeInfoDb/man/seqlevelsStyle.html'>mapSeqlevels</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html'>seqlengths</a></span><span class='op'>(</span><span class='va'>regions</span><span class='op'>)</span><span class='op'>)</span>, <span class='st'>"UCSC"</span><span class='op'>)</span></span>
<span class='r-in'><span class='op'>]</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## Get the region coverage</span></span>
<span class='r-in'><span class='va'>regionCov</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/derfinder/man/getRegionCoverage.html'>getRegionCoverage</a></span><span class='op'>(</span>fullCov <span class='op'>=</span> <span class='va'>fullCov</span>, regions <span class='op'>=</span> <span class='va'>regions</span><span class='op'>)</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> extendedMapSeqlevels: sequence names mapped from NCBI to UCSC for species homo_sapiens</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-08-10 01:23:03 getRegionCoverage: processing chr21</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-08-10 01:23:03 getRegionCoverage: done processing chr21</span>
<span class='r-in'><span class='co'>#</span></span>
<span class='r-in'><span class='co'>## Make plots for the regions</span></span>
<span class='r-in'><span class='fu'>plotRegionCoverage</span><span class='op'>(</span></span>
<span class='r-in'>    regions <span class='op'>=</span> <span class='va'>regions</span>, regionCoverage <span class='op'>=</span> <span class='va'>regionCov</span>,</span>
<span class='r-in'>    groupInfo <span class='op'>=</span> <span class='va'>genomeInfo</span><span class='op'>$</span><span class='va'>pop</span>, nearestAnnotation <span class='op'>=</span> <span class='va'>nearestAnnotation</span>,</span>
<span class='r-in'>    annotatedRegions <span class='op'>=</span> <span class='va'>annotatedRegions</span>, whichRegions <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>2</span></span>
<span class='r-in'><span class='op'>)</span></span>
<span class='r-plt'><img src='plotRegionCoverage-1.png' alt='' width='700' height='433' /></span>
<span class='r-plt'><img src='plotRegionCoverage-2.png' alt='' width='700' height='433' /></span>
<span class='r-plt'><img src='plotRegionCoverage-3.png' alt='' width='700' height='433' /></span>
<span class='r-in'></span>
<span class='r-in'><span class='co'>## Re-make plots with transcript information</span></span>
<span class='r-in'><span class='fu'>plotRegionCoverage</span><span class='op'>(</span></span>
<span class='r-in'>    regions <span class='op'>=</span> <span class='va'>regions</span>, regionCoverage <span class='op'>=</span> <span class='va'>regionCov</span>,</span>
<span class='r-in'>    groupInfo <span class='op'>=</span> <span class='va'>genomeInfo</span><span class='op'>$</span><span class='va'>pop</span>, nearestAnnotation <span class='op'>=</span> <span class='va'>nearestAnnotation</span>,</span>
<span class='r-in'>    annotatedRegions <span class='op'>=</span> <span class='va'>annotatedRegions</span>, whichRegions <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>2</span>,</span>
<span class='r-in'>    txdb <span class='op'>=</span> <span class='va'>TxDb.Hsapiens.UCSC.hg19.knownGene</span></span>
<span class='r-in'><span class='op'>)</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-08-10 01:23:03 plotRegionCoverage: extracting Tx info</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> 2021-08-10 01:23:06 plotRegionCoverage: getting Tx plot info</span>
<span class='r-plt'><img src='plotRegionCoverage-4.png' alt='' width='700' height='433' /></span>
<span class='r-plt'><img src='plotRegionCoverage-5.png' alt='' width='700' height='433' /></span>
<span class='r-plt'><img src='plotRegionCoverage-6.png' alt='' width='700' height='433' /></span>
<span class='r-in'><span class='kw'>if</span> <span class='op'>(</span><span class='cn'>FALSE</span><span class='op'>)</span> <span class='op'>{</span></span>
<span class='r-in'><span class='co'>## If you prefer, you can save the plots to a pdf file</span></span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/r/grDevices/pdf.html'>pdf</a></span><span class='op'>(</span><span class='st'>"ders.pdf"</span>, h <span class='op'>=</span> <span class='fl'>6</span>, w <span class='op'>=</span> <span class='fl'>9</span><span class='op'>)</span></span>
<span class='r-in'><span class='fu'>plotRegionCoverage</span><span class='op'>(</span></span>
<span class='r-in'>    regions <span class='op'>=</span> <span class='va'>regions</span>, regionCoverage <span class='op'>=</span> <span class='va'>regionCov</span>,</span>
<span class='r-in'>    groupInfo <span class='op'>=</span> <span class='va'>genomeInfo</span><span class='op'>$</span><span class='va'>pop</span>, nearestAnnotation <span class='op'>=</span> <span class='va'>nearestAnnotation</span>,</span>
<span class='r-in'>    annotatedRegions <span class='op'>=</span> <span class='va'>annotatedRegions</span>, whichRegions <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>2</span>,</span>
<span class='r-in'>    txdb <span class='op'>=</span> <span class='va'>TxDb.Hsapiens.UCSC.hg19.knownGene</span>, ask <span class='op'>=</span> <span class='cn'>FALSE</span></span>
<span class='r-in'><span class='op'>)</span></span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/r/grDevices/dev.html'>dev.off</a></span><span class='op'>(</span><span class='op'>)</span></span>
<span class='r-in'><span class='op'>}</span></span>
<span class='r-in'></span>
</code></pre></div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p><p>Developed by Leonardo Collado-Torres, Andrew E. Jaffe, Jeffrey T. Leek.</p></p>
</div>

<div class="pkgdown">
  <p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.6.1.9001.</p></p>
</div>

      </footer>
   </div>

  


  

  </body>
</html>


