<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Plot the coverage information surrounding a region cluster — plotCluster • derfinderPlot</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Plot the coverage information surrounding a region cluster — plotCluster" />
<meta property="og:description" content="For a given region found in calculatePvalues, plot the
coverage for the cluster this region belongs to as well as some padding. The
mean by group is shown to facilitate comparisons between groups. If
annotation exists, you can plot the trancripts and exons (if any)
overlapping in the vicinity of the region of interest." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">derfinderPlot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.21.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/derfinderPlot.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/leekgroup/derfinderPlot/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot the coverage information surrounding a region cluster</h1>
    <small class="dont-index">Source: <a href='https://github.com/leekgroup/derfinderPlot/blob/master/R/plotCluster.R'><code>R/plotCluster.R</code></a></small>
    <div class="hidden name"><code>plotCluster.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>For a given region found in <a href='https://rdrr.io/pkg/derfinder/man/calculatePvalues.html'>calculatePvalues</a>, plot the
coverage for the cluster this region belongs to as well as some padding. The
mean by group is shown to facilitate comparisons between groups. If
annotation exists, you can plot the trancripts and exons (if any)
overlapping in the vicinity of the region of interest.</p>
    </div>

    <pre class="usage"><span class='fu'>plotCluster</span>(
  <span class='no'>idx</span>,
  <span class='no'>regions</span>,
  <span class='no'>annotation</span>,
  <span class='no'>coverageInfo</span>,
  <span class='no'>groupInfo</span>,
  <span class='kw'>titleUse</span> <span class='kw'>=</span> <span class='st'>"qval"</span>,
  <span class='kw'>txdb</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>p.ideogram</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='no'>...</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>idx</th>
      <td><p>A integer specifying the index number of the region of interest.
This region is graphically highlighted by a red bar.</p></td>
    </tr>
    <tr>
      <th>regions</th>
      <td><p>The <code>$regions</code> output from
<a href='https://rdrr.io/pkg/derfinder/man/calculatePvalues.html'>calculatePvalues</a>.</p></td>
    </tr>
    <tr>
      <th>annotation</th>
      <td><p>The output from running <a href='https://rdrr.io/pkg/bumphunter/man/matchGenes.html'>matchGenes</a>
on the output from <a href='https://rdrr.io/pkg/derfinder/man/calculatePvalues.html'>calculatePvalues</a>.</p></td>
    </tr>
    <tr>
      <th>coverageInfo</th>
      <td><p>A DataFrame resulting from
<a href='https://rdrr.io/pkg/derfinder/man/loadCoverage.html'>loadCoverage</a> using <code>cutoff=NULL</code>.</p></td>
    </tr>
    <tr>
      <th>groupInfo</th>
      <td><p>A factor specifying the group membership of each sample. It
will be used to color the samples by group.</p></td>
    </tr>
    <tr>
      <th>titleUse</th>
      <td><p>Whether to show the p-value (<code>pval</code>), the q-value
(<code>qval</code>) or the FWER adjusted p-value (<code>fwer</code>) in the title. If
<code>titleUse=none</code> then no p-value or
q-value information is used; useful if no permutations were performed and
thus p-value and q-value information is absent.</p></td>
    </tr>
    <tr>
      <th>txdb</th>
      <td><p>A transcript data base such as
<code>TxDb.Hsapiens.UCSC.hg19.knownGene</code>. If <code>NULL</code> then no annotation
information is used.</p></td>
    </tr>
    <tr>
      <th>p.ideogram</th>
      <td><p>If <code>NULL</code>, the ideogram for hg19 is built for the
corresponding chromosome. Otherwise an ideogram resuling from
plotIdeogram.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Arguments passed to other methods and/or advanced arguments.
Advanced arguments:</p><dl'>
<dt>maxExtend </dt><dd><p>The maximum number of base-pairs to extend the view (on
each side) before and after the region cluster of interest. For small region
clusters, the one side extension is equal to the width of the region
cluster.</p></dd>
<dt>colsubset </dt><dd><p>Column subset in case that it was specified in
<a href='https://rdrr.io/pkg/derfinder/man/preprocessCoverage.html'>preprocessCoverage</a>.</p></dd>
<dt>forceLarge </dt><dd><p>If <code>TRUE</code> then the data size limitations are
ignored.
The window size (region cluster width + 2 times <code>maxExtend</code>) has to be
less than 100 kb. Note that a single plot at the 300kb range can take around
2 hours to complete.</p></dd>

</dl></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A ggplot2 plot that is ready to be printed out. Tecnically it is a
ggbio object. The region with the red bar is the one whose information is
shown in the title.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>See the parameter <code>significantCut</code> in
<a href='https://rdrr.io/pkg/derfinder/man/calculatePvalues.html'>calculatePvalues</a> for how the significance cutoffs are
determined.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><a href='https://rdrr.io/pkg/derfinder/man/loadCoverage.html'>loadCoverage</a>, <a href='https://rdrr.io/pkg/derfinder/man/calculatePvalues.html'>calculatePvalues</a>,
<a href='https://rdrr.io/pkg/bumphunter/man/matchGenes.html'>matchGenes</a>, plotIdeogram</p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>## Load data</span>
<span class='fu'><a href='https://rdrr.io/r/base/library.html'>library</a></span>(<span class='st'>"derfinder"</span>)

<span class='co'>## Annotate the results with bumphunter::matchGenes()</span>
<span class='fu'><a href='https://rdrr.io/r/base/library.html'>library</a></span>(<span class='st'>"bumphunter"</span>)</div><div class='output co'>#&gt; <span class='message'>Loading required package: S4Vectors</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: stats4</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: BiocGenerics</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: parallel</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: 'BiocGenerics'</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from 'package:parallel':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
#&gt; <span class='message'>    clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
#&gt; <span class='message'>    parLapplyLB, parRapply, parSapply, parSapplyLB</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from 'package:stats':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    IQR, mad, sd, var, xtabs</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from 'package:base':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
#&gt; <span class='message'>    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
#&gt; <span class='message'>    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
#&gt; <span class='message'>    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
#&gt; <span class='message'>    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
#&gt; <span class='message'>    union, unique, unsplit, which, which.max, which.min</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: 'S4Vectors'</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from 'package:base':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    expand.grid</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: IRanges</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomeInfoDb</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomicRanges</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: foreach</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: iterators</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: locfit</span></div><div class='output co'>#&gt; <span class='message'>locfit 1.5-9.4 	 2020-03-24</span></div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/library.html'>library</a></span>(<span class='st'>"TxDb.Hsapiens.UCSC.hg19.knownGene"</span>)</div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomicFeatures</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: AnnotationDbi</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: Biobase</span></div><div class='output co'>#&gt; <span class='message'>Welcome to Bioconductor</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    Vignettes contain introductory material; view with</span>
#&gt; <span class='message'>    'browseVignettes()'. To cite Bioconductor, see</span>
#&gt; <span class='message'>    'citation("Biobase")', and for packages 'citation("pkgname")'.</span></div><div class='input'><span class='fu'><a href='https://rdrr.io/r/base/library.html'>library</a></span>(<span class='st'>"org.Hs.eg.db"</span>)</div><div class='output co'>#&gt; </div><div class='input'><span class='no'>genes</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/bumphunter/man/annotateTranscripts.html'>annotateTranscripts</a></span>(
    <span class='kw'>txdb</span> <span class='kw'>=</span> <span class='no'>TxDb.Hsapiens.UCSC.hg19.knownGene</span>,
    <span class='kw'>annotationPackage</span> <span class='kw'>=</span> <span class='st'>"org.Hs.eg.db"</span>
)</div><div class='output co'>#&gt; <span class='message'>Getting TSS and TSE.</span></div><div class='output co'>#&gt; <span class='message'>Getting CSS and CSE.</span></div><div class='output co'>#&gt; <span class='message'>Getting exons.</span></div><div class='output co'>#&gt; <span class='message'>Annotating genes.</span></div><div class='input'><span class='no'>annotation</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/bumphunter/man/matchGenes.html'>matchGenes</a></span>(<span class='kw'>x</span> <span class='kw'>=</span> <span class='no'>genomeRegions</span>$<span class='no'>regions</span>, <span class='kw'>subject</span> <span class='kw'>=</span> <span class='no'>genes</span>)

<span class='co'>## Make the plot</span>
<span class='fu'>plotCluster</span>(
    <span class='kw'>idx</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>regions</span> <span class='kw'>=</span> <span class='no'>genomeRegions</span>$<span class='no'>regions</span>, <span class='kw'>annotation</span> <span class='kw'>=</span> <span class='no'>annotation</span>,
    <span class='kw'>coverageInfo</span> <span class='kw'>=</span> <span class='no'>genomeDataRaw</span>$<span class='no'>coverage</span>, <span class='kw'>groupInfo</span> <span class='kw'>=</span> <span class='no'>genomeInfo</span>$<span class='no'>pop</span>,
    <span class='kw'>txdb</span> <span class='kw'>=</span> <span class='no'>TxDb.Hsapiens.UCSC.hg19.knownGene</span>
)</div><div class='output co'>#&gt; <span class='warning'>Warning: geom(ideogram) need valid seqlengths information for accurate mapping,</span>
#&gt; <span class='warning'>                 now use reduced information as ideogram... </span></div><div class='output co'>#&gt; <span class='warning'>Warning: `panel.margin` is deprecated. Please use `panel.spacing` property instead</span></div><div class='output co'>#&gt; <span class='warning'>Warning: `panel.margin` is deprecated. Please use `panel.spacing` property instead</span></div><div class='output co'>#&gt; <span class='message'>Parsing transcripts...</span></div><div class='output co'>#&gt; <span class='message'>Parsing exons...</span></div><div class='output co'>#&gt; <span class='message'>Parsing cds...</span></div><div class='output co'>#&gt; <span class='message'>Parsing utrs...</span></div><div class='output co'>#&gt; <span class='message'>------exons...</span></div><div class='output co'>#&gt; <span class='message'>------cdss...</span></div><div class='output co'>#&gt; <span class='message'>------introns...</span></div><div class='output co'>#&gt; <span class='message'>------utr...</span></div><div class='output co'>#&gt; <span class='message'>aggregating...</span></div><div class='output co'>#&gt; <span class='message'>Done</span></div><div class='output co'>#&gt; <span class='message'>Constructing graphics...</span></div><div class='output co'>#&gt; <span class='warning'>Warning: `quo_expr()` is deprecated as of rlang 0.2.0.</span>
#&gt; <span class='warning'>Please use `quo_squash()` instead.</span>
#&gt; <span class='warning'>This warning is displayed once per session.</span></div><div class='output co'>#&gt; <span class='warning'>Warning: `panel.margin` is deprecated. Please use `panel.spacing` property instead</span></div><div class='output co'>#&gt; <span class='warning'>Warning: `panel.margin` is deprecated. Please use `panel.spacing` property instead</span></div><div class='output co'>#&gt; <span class='warning'>Warning: `panel.margin` is deprecated. Please use `panel.spacing` property instead</span></div><div class='output co'>#&gt; <span class='warning'>Warning: `panel.margin` is deprecated. Please use `panel.spacing` property instead</span></div><div class='output co'>#&gt; <span class='warning'>Warning: `panel.margin` is deprecated. Please use `panel.spacing` property instead</span></div><div class='output co'>#&gt; <span class='warning'>Warning: `panel.margin` is deprecated. Please use `panel.spacing` property instead</span></div><div class='output co'>#&gt; <span class='warning'>Warning: `panel.margin` is deprecated. Please use `panel.spacing` property instead</span></div><div class='output co'>#&gt; <span class='warning'>Warning: `panel.margin` is deprecated. Please use `panel.spacing` property instead</span></div><div class='output co'>#&gt; <span class='warning'>Warning: `panel.margin` is deprecated. Please use `panel.spacing` property instead</span></div><div class='output co'>#&gt; <span class='message'>Coordinate system already present. Adding new coordinate system, which will replace the existing one.</span></div><div class='output co'>#&gt; <span class='message'>Coordinate system already present. Adding new coordinate system, which will replace the existing one.</span></div><div class='output co'>#&gt; <span class='message'>Coordinate system already present. Adding new coordinate system, which will replace the existing one.</span></div><div class='output co'>#&gt; <span class='message'>Coordinate system already present. Adding new coordinate system, which will replace the existing one.</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Transformation introduced infinite values in continuous y-axis</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Transformation introduced infinite values in continuous y-axis</span></div><div class='img'><img src='plotCluster-1.png' alt='' width='700' height='433' /></div><div class='input'><span class='co'>## Resize the plot window and the labels will look good.</span>
<span class='kw'>if</span> (<span class='fl'>FALSE</span>) {
<span class='co'>## For a custom plot, check the ggbio and ggplot2 packages.</span>
<span class='co'>## Also feel free to look at the code for this function:</span>
<span class='no'>plotCluster</span>
}</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Leonardo Collado-Torres, Andrew E. Jaffe, Jeffrey T. Leek.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


