<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to derfinderPlot • derfinderPlot</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to derfinderPlot">
<meta property="og:description" content="derfinderPlot">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">derfinderPlot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.35.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/derfinderPlot.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/leekgroup/derfinderPlot/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to derfinderPlot</h1>
                        <h4 data-toc-skip class="author">Leonardo
Collado-Torres</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical
CampusCenter for Computational Biology, Johns Hopkins
University<br><a class="author_email" href="mailto:#"></a><a href="mailto:lcolladotor@gmail.com" class="email">lcolladotor@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">7 May 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/leekgroup/derfinderPlot/blob/HEAD/vignettes/derfinderPlot.Rmd" class="external-link"><code>vignettes/derfinderPlot.Rmd</code></a></small>
      <div class="hidden name"><code>derfinderPlot.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="basics">Basics<a class="anchor" aria-label="anchor" href="#basics"></a>
</h2>
<div class="section level3">
<h3 id="install-derfinderplot">Install <em><a href="https://bioconductor.org/packages/3.17/derfinderPlot" class="external-link">derfinderPlot</a></em><a class="anchor" aria-label="anchor" href="#install-derfinderplot"></a>
</h3>
<p><code>R</code> is an open-source statistical environment which can be
easily modified to enhance its functionality via packages. <em><a href="https://bioconductor.org/packages/3.17/derfinderPlot" class="external-link">derfinderPlot</a></em>
is a <code>R</code> package available via the <a href="http://bioconductor/packages/derfinderPlot" class="external-link">Bioconductor</a>
repository for packages. <code>R</code> can be installed on any
operating system from <a href="https://cran.r-project.org/" class="external-link">CRAN</a>
after which you can install <em><a href="https://bioconductor.org/packages/3.17/derfinderPlot" class="external-link">derfinderPlot</a></em>
by using the following commands in your <code>R</code> session:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"derfinderPlot"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Check that you have a valid Bioconductor installation</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/valid.html" class="external-link">valid</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="required-knowledge">Required knowledge<a class="anchor" aria-label="anchor" href="#required-knowledge"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.17/derfinderPlot" class="external-link">derfinderPlot</a></em>
is based on many other packages and in particular in those that have
implemented the infrastructure needed for dealing with RNA-seq data. A
<em><a href="https://bioconductor.org/packages/3.17/derfinderPlot" class="external-link">derfinderPlot</a></em>
user is not expected to deal with those packages directly but will need
to be familiar with <em><a href="https://bioconductor.org/packages/3.17/derfinder" class="external-link">derfinder</a></em>
and for some plots with <em><a href="https://bioconductor.org/packages/3.17/ggbio" class="external-link">ggbio</a></em>.</p>
<p>If you are asking yourself the question “Where do I start using
Bioconductor?” you might be interested in <a href="http://lcolladotor.github.io/2014/10/16/startbioc/#.VkOKbq6rRuU" class="external-link">this
blog post</a>.</p>
</div>
<div class="section level3">
<h3 id="asking-for-help">Asking for help<a class="anchor" aria-label="anchor" href="#asking-for-help"></a>
</h3>
<p>As package developers, we try to explain clearly how to use our
packages and in which order to use the functions. But <code>R</code> and
<code>Bioconductor</code> have a steep learning curve so it is critical
to learn where to ask for help. The blog post quoted above mentions some
but we would like to highlight the <a href="https://support.bioconductor.org/" class="external-link">Bioconductor support site</a>
as the main resource for getting help: remember to use the
<code>derfinder</code> or <code>derfinderPlot</code> tags and check <a href="https://support.bioconductor.org/t/derfinder/" class="external-link">the older
posts</a>. Other alternatives are available such as creating GitHub
issues and tweeting. However, please note that if you want to receive
help you should adhere to the <a href="http://www.bioconductor.org/help/support/posting-guide/" class="external-link">posting
guidelines</a>. It is particularly critical that you provide a small
reproducible example and your session information so package developers
can track down the source of the error.</p>
</div>
<div class="section level3">
<h3 id="citing-derfinderplot">Citing <em><a href="https://bioconductor.org/packages/3.17/derfinderPlot" class="external-link">derfinderPlot</a></em><a class="anchor" aria-label="anchor" href="#citing-derfinderplot"></a>
</h3>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.17/derfinderPlot" class="external-link">derfinderPlot</a></em>
will be useful for your research. Please use the following information
to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Citation info</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"derfinderPlot"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## To cite package 'derfinderPlot' in publications use:</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   Collado-Torres L, Jaffe AE, Leek JT (2017). _derfinderPlot: Plotting</span></span>
<span><span class="co">##   functions for derfinder_. doi:10.18129/B9.bioc.derfinderPlot</span></span>
<span><span class="co">##   &lt;https://doi.org/10.18129/B9.bioc.derfinderPlot&gt;,</span></span>
<span><span class="co">##   https://github.com/leekgroup/derfinderPlot - R package version</span></span>
<span><span class="co">##   1.35.0, &lt;http://www.bioconductor.org/packages/derfinderPlot&gt;.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   Collado-Torres L, Nellore A, Frazee AC, Wilks C, Love MI, Langmead B,</span></span>
<span><span class="co">##   Irizarry RA, Leek JT, Jaffe AE (2017). "Flexible expressed region</span></span>
<span><span class="co">##   analysis for RNA-seq with derfinder." _Nucl. Acids Res._.</span></span>
<span><span class="co">##   doi:10.1093/nar/gkw852 &lt;https://doi.org/10.1093/nar/gkw852&gt;,</span></span>
<span><span class="co">##   &lt;http://nar.oxfordjournals.org/content/early/2016/09/29/nar.gkw852&gt;.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## To see these entries in BibTeX format, use 'print(&lt;citation&gt;,</span></span>
<span><span class="co">## bibtex=TRUE)', 'toBibtex(.)', or set</span></span>
<span><span class="co">## 'options(citation.bibtex.max=999)'.</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="introduction-to-derfinderplot">Introduction to <em><a href="https://bioconductor.org/packages/3.17/derfinderPlot" class="external-link">derfinderPlot</a></em><a class="anchor" aria-label="anchor" href="#introduction-to-derfinderplot"></a>
</h2>
<p><em><a href="https://bioconductor.org/packages/3.17/derfinderPlot" class="external-link">derfinderPlot</a></em>
(Collado-Torres, Jaffe, and Leek, 2017) is an addon package for <em><a href="https://bioconductor.org/packages/3.17/derfinder" class="external-link">derfinder</a></em>
(Collado-Torres, Nellore, Frazee, Wilks, Love, Langmead, Irizarry, Leek,
and Jaffe, 2017) with functions that allow you to visualize the
results.</p>
<p>While the functions in <em><a href="https://bioconductor.org/packages/3.17/derfinderPlot" class="external-link">derfinderPlot</a></em>
assume you generated the data with <em><a href="https://bioconductor.org/packages/3.17/derfinder" class="external-link">derfinder</a></em>,
they can be used with other <code>GRanges</code> objects properly
formatted.</p>
<p>The functions in <em><a href="https://bioconductor.org/packages/3.17/derfinderPlot" class="external-link">derfinderPlot</a></em>
are:</p>
<ul>
<li>
<code><a href="../reference/plotCluster.html">plotCluster()</a></code> is a tailored <em><a href="https://bioconductor.org/packages/3.17/ggbio" class="external-link">ggbio</a></em> (Yin,
Cook, and Lawrence, 2012) plot that shows all the regions in a cluster
(defined by distance). It shows the base-level coverage for each sample
as well as the mean for each group. If these regions overlap any known
gene, the gene and the transcript annotation is displayed.</li>
<li>
<code><a href="../reference/plotOverview.html">plotOverview()</a></code> is another tailored <em><a href="https://bioconductor.org/packages/3.17/ggbio" class="external-link">ggbio</a></em> (Yin,
Cook, and Lawrence, 2012) plot showing an overview of the whole genome.
This plot can be useful to observe if the regions are clustered in a
subset of a chromosome. It can also be used to check whether the regions
match predominantly one part of the gene structure (for example, 3’
overlaps).</li>
<li>
<code><a href="../reference/plotRegionCoverage.html">plotRegionCoverage()</a></code> is a fast plotting function using
<code>R</code> base graphics that shows the base-level coverage for each
sample inside a specific region of the genome. If the region overlaps
any known gene or intron, the information is displayed. Optionally, it
can display the known transcripts. This function is most likely the
easiest to use with <code>GRanges</code> objects from other
packages.</li>
</ul>
</div>
<div class="section level2">
<h2 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h2>
<p>As an example, we will analyze a small subset of the samples from the
<em>BrainSpan Atlas of the Human Brain</em> (BrainSpan, 2011) publicly
available data.</p>
<p>We first load the required packages.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load libraries</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/lcolladotor/derfinder" class="external-link">"derfinder"</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/leekgroup/derfinderData" class="external-link">"derfinderData"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/leekgroup/derfinderPlot" class="external-link">"derfinderPlot"</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Registered S3 method overwritten by 'GGally':</span></span>
<span><span class="co">##   method from   </span></span>
<span><span class="co">##   +.gg   ggplot2</span></span></code></pre>
<div class="section level3">
<h3 id="analyze-data">Analyze data<a class="anchor" aria-label="anchor" href="#analyze-data"></a>
</h3>
<p>For this example, we created a small table with the relevant
phenotype data for 12 samples: 6 from fetal samples and 6 from adult
samples. We chose at random a brain region, in this case the primary
auditory cortex (core) and for the example we will only look at data
from chromosome 21. Other variables include the age in years and the
gender. The data is shown below.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://yihui.org/knitr/" class="external-link">"knitr"</a></span><span class="op">)</span></span>
<span><span class="co">## Get pheno table</span></span>
<span><span class="va">pheno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">brainspanPheno</span>, <span class="va">structure_acronym</span> <span class="op">==</span> <span class="st">"A1C"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Display the main information</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">pheno</span><span class="op">[</span>, <span class="op">-</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">pheno</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"structure_acronym"</span>,</span>
<span>    <span class="st">"structure_name"</span>, <span class="st">"file"</span></span>
<span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">p</span>, format <span class="op">=</span> <span class="st">"html"</span>, row.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
gender
</th>
<th style="text-align:left;">
lab
</th>
<th style="text-align:right;">
Age
</th>
<th style="text-align:left;">
group
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
HSB114.A1C
</td>
<td style="text-align:right;">
-0.5192308
</td>
<td style="text-align:left;">
fetal
</td>
</tr>
<tr>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
HSB103.A1C
</td>
<td style="text-align:right;">
-0.5192308
</td>
<td style="text-align:left;">
fetal
</td>
</tr>
<tr>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
HSB178.A1C
</td>
<td style="text-align:right;">
-0.4615385
</td>
<td style="text-align:left;">
fetal
</td>
</tr>
<tr>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
HSB154.A1C
</td>
<td style="text-align:right;">
-0.4615385
</td>
<td style="text-align:left;">
fetal
</td>
</tr>
<tr>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
F
</td>
<td style="text-align:left;">
HSB150.A1C
</td>
<td style="text-align:right;">
-0.5384615
</td>
<td style="text-align:left;">
fetal
</td>
</tr>
<tr>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
F
</td>
<td style="text-align:left;">
HSB149.A1C
</td>
<td style="text-align:right;">
-0.5192308
</td>
<td style="text-align:left;">
fetal
</td>
</tr>
<tr>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
F
</td>
<td style="text-align:left;">
HSB130.A1C
</td>
<td style="text-align:right;">
21.0000000
</td>
<td style="text-align:left;">
adult
</td>
</tr>
<tr>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
HSB136.A1C
</td>
<td style="text-align:right;">
23.0000000
</td>
<td style="text-align:left;">
adult
</td>
</tr>
<tr>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
F
</td>
<td style="text-align:left;">
HSB126.A1C
</td>
<td style="text-align:right;">
30.0000000
</td>
<td style="text-align:left;">
adult
</td>
</tr>
<tr>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
HSB145.A1C
</td>
<td style="text-align:right;">
36.0000000
</td>
<td style="text-align:left;">
adult
</td>
</tr>
<tr>
<td style="text-align:left;">
11
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
HSB123.A1C
</td>
<td style="text-align:right;">
37.0000000
</td>
<td style="text-align:left;">
adult
</td>
</tr>
<tr>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
F
</td>
<td style="text-align:left;">
HSB135.A1C
</td>
<td style="text-align:right;">
40.0000000
</td>
<td style="text-align:left;">
adult
</td>
</tr>
</tbody>
</table>
<p>We can load the data from <em><a href="https://bioconductor.org/packages/3.17/derfinderData" class="external-link">derfinderData</a></em>
(Collado-Torres, Jaffe, and Leek, 2023) by first identifying the paths
to the BigWig files with <code><a href="https://rdrr.io/pkg/derfinder/man/rawFiles.html" class="external-link">derfinder::rawFiles()</a></code> and then
loading the data with <code><a href="https://rdrr.io/pkg/derfinder/man/fullCoverage.html" class="external-link">derfinder::fullCoverage()</a></code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Determine the files to use and fix the names</span></span>
<span><span class="va">files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/rawFiles.html" class="external-link">rawFiles</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"A1C"</span>, package <span class="op">=</span> <span class="st">"derfinderData"</span><span class="op">)</span>,</span>
<span>    samplepatt <span class="op">=</span> <span class="st">"bw"</span>, fileterm <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">files</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">".bw"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">files</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Load the data from disk</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="va">fullCov</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/fullCoverage.html" class="external-link">fullCoverage</a></span><span class="op">(</span>files <span class="op">=</span> <span class="va">files</span>, chrs <span class="op">=</span> <span class="st">"chr21"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 2023-05-07 01:31:26.391206 fullCoverage: processing chromosome chr21</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:26.408351 loadCoverage: finding chromosome lengths</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:26.44216 loadCoverage: loading BigWig file /__w/_temp/Library/derfinderData/extdata/A1C/HSB103.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:26.663699 loadCoverage: loading BigWig file /__w/_temp/Library/derfinderData/extdata/A1C/HSB114.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:26.854328 loadCoverage: loading BigWig file /__w/_temp/Library/derfinderData/extdata/A1C/HSB123.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:27.002744 loadCoverage: loading BigWig file /__w/_temp/Library/derfinderData/extdata/A1C/HSB126.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:27.104128 loadCoverage: loading BigWig file /__w/_temp/Library/derfinderData/extdata/A1C/HSB130.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:27.226225 loadCoverage: loading BigWig file /__w/_temp/Library/derfinderData/extdata/A1C/HSB135.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:27.326831 loadCoverage: loading BigWig file /__w/_temp/Library/derfinderData/extdata/A1C/HSB136.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:27.440618 loadCoverage: loading BigWig file /__w/_temp/Library/derfinderData/extdata/A1C/HSB145.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:27.55806 loadCoverage: loading BigWig file /__w/_temp/Library/derfinderData/extdata/A1C/HSB149.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:27.696518 loadCoverage: loading BigWig file /__w/_temp/Library/derfinderData/extdata/A1C/HSB150.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:27.803245 loadCoverage: loading BigWig file /__w/_temp/Library/derfinderData/extdata/A1C/HSB154.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:27.944826 loadCoverage: loading BigWig file /__w/_temp/Library/derfinderData/extdata/A1C/HSB178.bw</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:28.0956 loadCoverage: applying the cutoff to the merged data</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:28.131841 filterData: originally there were 48129895 rows, now there are 48129895 rows. Meaning that 0 percent was filtered.</span></span></code></pre>
<pre><code><span><span class="co">##    user  system elapsed </span></span>
<span><span class="co">##   1.774   0.036   1.810</span></span></code></pre>
<p>Alternatively, since the BigWig files are publicly available from
<em>BrainSpan</em> (see <a href="http://download.alleninstitute.org/brainspan/MRF_BigWig_Gencode_v10/bigwig/" class="external-link">here</a>),
we can extract the relevant coverage data using
<code><a href="https://rdrr.io/pkg/derfinder/man/fullCoverage.html" class="external-link">derfinder::fullCoverage()</a></code>. Note that as of <em><a href="https://bioconductor.org/packages/3.17/rtracklayer" class="external-link">rtracklayer</a></em>
1.25.16 BigWig files are not supported on Windows: you can find the
<code>fullCov</code> object inside <em><a href="https://bioconductor.org/packages/3.17/derfinderData" class="external-link">derfinderData</a></em>
to follow the examples.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Determine the files to use and fix the names</span></span>
<span><span class="va">files</span> <span class="op">&lt;-</span> <span class="va">pheno</span><span class="op">$</span><span class="va">file</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">files</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">".A1C"</span>, <span class="st">""</span>, <span class="va">pheno</span><span class="op">$</span><span class="va">lab</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Load the data from the web</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="va">fullCov</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/fullCoverage.html" class="external-link">fullCoverage</a></span><span class="op">(</span>files <span class="op">=</span> <span class="va">files</span>, chrs <span class="op">=</span> <span class="st">"chr21"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Once we have the base-level coverage data for all 12 samples, we can
construct the models. In this case, we want to find differences between
fetal and adult samples while adjusting for gender and a proxy of the
library size.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Get some idea of the library sizes</span></span>
<span><span class="va">sampleDepths</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/sampleDepth.html" class="external-link">sampleDepth</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/collapseFullCoverage.html" class="external-link">collapseFullCoverage</a></span><span class="op">(</span><span class="va">fullCov</span><span class="op">)</span>, <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 2023-05-07 01:31:28.394561 sampleDepth: Calculating sample quantiles</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:28.533602 sampleDepth: Calculating sample adjustments</span></span></code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Define models</span></span>
<span><span class="va">models</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/makeModels.html" class="external-link">makeModels</a></span><span class="op">(</span><span class="va">sampleDepths</span>,</span>
<span>    testvars <span class="op">=</span> <span class="va">pheno</span><span class="op">$</span><span class="va">group</span>,</span>
<span>    adjustvars <span class="op">=</span> <span class="va">pheno</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gender"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Next, we can find candidate differentially expressed regions (DERs)
using as input the segments of the genome where at least one sample has
coverage greater than 3. In this particular example, we chose a low
theoretical F-statistic cutoff and used 20 permutations.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Filter coverage</span></span>
<span><span class="va">filteredCov</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">fullCov</span>, <span class="va">filterData</span>, cutoff <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 2023-05-07 01:31:28.880557 filterData: originally there were 48129895 rows, now there are 90023 rows. Meaning that 99.81 percent was filtered.</span></span></code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Perform differential expression analysis</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/rafalab/bumphunter" class="external-link">"bumphunter"</a></span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/analyzeChr.html" class="external-link">analyzeChr</a></span><span class="op">(</span></span>
<span>    chr <span class="op">=</span> <span class="st">"chr21"</span>, <span class="va">filteredCov</span><span class="op">$</span><span class="va">chr21</span>,</span>
<span>    <span class="va">models</span>, groupInfo <span class="op">=</span> <span class="va">pheno</span><span class="op">$</span><span class="va">group</span>, writeOutput <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    cutoffFstat <span class="op">=</span> <span class="fl">5e-02</span>, nPermute <span class="op">=</span> <span class="fl">20</span>, seeds <span class="op">=</span> <span class="fl">20140923</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">20</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 2023-05-07 01:31:30.053725 analyzeChr: Pre-processing the coverage data</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:31.633721 analyzeChr: Calculating statistics</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:31.63681 calculateStats: calculating the F-statistics</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:31.819212 analyzeChr: Calculating pvalues</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:31.81955 analyzeChr: Using the following theoretical cutoff for the F-statistics 5.31765507157871</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:31.820765 calculatePvalues: identifying data segments</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:31.82819 findRegions: segmenting information</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:32.83936 findRegions: identifying candidate regions</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:32.893461 findRegions: identifying region clusters</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:33.04174 calculatePvalues: calculating F-statistics for permutation 1 and seed 20140924</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:33.181276 findRegions: segmenting information</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:33.203026 findRegions: identifying candidate regions</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:33.263142 calculatePvalues: calculating F-statistics for permutation 2 and seed 20140925</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:33.420655 findRegions: segmenting information</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:33.442395 findRegions: identifying candidate regions</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:33.486974 calculatePvalues: calculating F-statistics for permutation 3 and seed 20140926</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:33.626868 findRegions: segmenting information</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:33.648775 findRegions: identifying candidate regions</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:33.703494 calculatePvalues: calculating F-statistics for permutation 4 and seed 20140927</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:33.842633 findRegions: segmenting information</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:33.864444 findRegions: identifying candidate regions</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:33.909625 calculatePvalues: calculating F-statistics for permutation 5 and seed 20140928</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:34.064129 findRegions: segmenting information</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:34.086158 findRegions: identifying candidate regions</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:34.130708 calculatePvalues: calculating F-statistics for permutation 6 and seed 20140929</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:34.283694 findRegions: segmenting information</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:34.319327 findRegions: identifying candidate regions</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:34.363508 calculatePvalues: calculating F-statistics for permutation 7 and seed 20140930</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:34.503148 findRegions: segmenting information</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:34.525093 findRegions: identifying candidate regions</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:34.569224 calculatePvalues: calculating F-statistics for permutation 8 and seed 20140931</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:34.725616 findRegions: segmenting information</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:34.747399 findRegions: identifying candidate regions</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:34.791714 calculatePvalues: calculating F-statistics for permutation 9 and seed 20140932</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:34.930967 findRegions: segmenting information</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:34.952786 findRegions: identifying candidate regions</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:35.00683 calculatePvalues: calculating F-statistics for permutation 10 and seed 20140933</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:35.146759 findRegions: segmenting information</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:35.168657 findRegions: identifying candidate regions</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:35.213592 calculatePvalues: calculating F-statistics for permutation 11 and seed 20140934</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:35.36346 findRegions: segmenting information</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:35.385319 findRegions: identifying candidate regions</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:35.429254 calculatePvalues: calculating F-statistics for permutation 12 and seed 20140935</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:35.578584 findRegions: segmenting information</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:35.600233 findRegions: identifying candidate regions</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:35.643447 calculatePvalues: calculating F-statistics for permutation 13 and seed 20140936</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:35.792182 findRegions: segmenting information</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:35.813905 findRegions: identifying candidate regions</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:35.857596 calculatePvalues: calculating F-statistics for permutation 14 and seed 20140937</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:36.017794 findRegions: segmenting information</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:36.03957 findRegions: identifying candidate regions</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:36.084647 calculatePvalues: calculating F-statistics for permutation 15 and seed 20140938</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:36.223148 findRegions: segmenting information</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:36.245006 findRegions: identifying candidate regions</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:36.288612 calculatePvalues: calculating F-statistics for permutation 16 and seed 20140939</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:36.440774 findRegions: segmenting information</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:36.462454 findRegions: identifying candidate regions</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:36.506334 calculatePvalues: calculating F-statistics for permutation 17 and seed 20140940</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:36.660499 findRegions: segmenting information</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:36.682193 findRegions: identifying candidate regions</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:36.725765 calculatePvalues: calculating F-statistics for permutation 18 and seed 20140941</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:36.877477 findRegions: segmenting information</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:36.899216 findRegions: identifying candidate regions</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:36.942913 calculatePvalues: calculating F-statistics for permutation 19 and seed 20140942</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:37.097921 findRegions: segmenting information</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:37.119517 findRegions: identifying candidate regions</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:37.162937 calculatePvalues: calculating F-statistics for permutation 20 and seed 20140943</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:37.314812 findRegions: segmenting information</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:37.336471 findRegions: identifying candidate regions</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:37.401037 calculatePvalues: calculating the p-values</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:31:37.466817 analyzeChr: Annotating regions</span></span></code></pre>
<pre><code><span><span class="co">## No annotationPackage supplied. Trying org.Hs.eg.db.</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: org.Hs.eg.db</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: AnnotationDbi</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: Biobase</span></span></code></pre>
<pre><code><span><span class="co">## Welcome to Bioconductor</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     Vignettes contain introductory material; view with</span></span>
<span><span class="co">##     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">##     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span></code></pre>
<pre><code><span><span class="co">## </span></span></code></pre>
<pre><code><span><span class="co">## Getting TSS and TSE.</span></span></code></pre>
<pre><code><span><span class="co">## Getting CSS and CSE.</span></span></code></pre>
<pre><code><span><span class="co">## Getting exons.</span></span></code></pre>
<pre><code><span><span class="co">## Annotating genes.</span></span></code></pre>
<pre><code><span><span class="co">## ...</span></span></code></pre>
<pre><code><span><span class="co">##    user  system elapsed </span></span>
<span><span class="co">##  54.266   1.836  53.147</span></span></code></pre>
<div class="sourceCode" id="cb116"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Quick access to the results</span></span>
<span><span class="va">regions</span> <span class="op">&lt;-</span> <span class="va">results</span><span class="op">$</span><span class="va">regions</span><span class="op">$</span><span class="va">regions</span></span>
<span></span>
<span><span class="co">## Annotation database to use</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"TxDb.Hsapiens.UCSC.hg19.knownGene"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">txdb</span> <span class="op">&lt;-</span> <span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="plotoverview">
<code>plotOverview()</code><a class="anchor" aria-label="anchor" href="#plotoverview"></a>
</h3>
<p>Now that we have obtained the main results using <em><a href="https://bioconductor.org/packages/3.17/derfinder" class="external-link">derfinder</a></em>,
we can proceed to visualizing the results using <em><a href="https://bioconductor.org/packages/3.17/derfinderPlot" class="external-link">derfinderPlot</a></em>.
The easiest to use of all the functions is <code><a href="../reference/plotOverview.html">plotOverview()</a></code>
which takes a set of regions and annotation information produced by
<code><a href="https://rdrr.io/pkg/bumphunter/man/matchGenes.html" class="external-link">bumphunter::matchGenes()</a></code>.</p>
<p>Figure @ref(fig:plotOverview) shows the candidate DERs colored by
whether their q-value was less than 0.10 or not.</p>
<div class="sourceCode" id="cb117"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Q-values overview</span></span>
<span><span class="fu"><a href="../reference/plotOverview.html">plotOverview</a></span><span class="op">(</span>regions <span class="op">=</span> <span class="va">regions</span>, annotation <span class="op">=</span> <span class="va">results</span><span class="op">$</span><span class="va">annotation</span>, type <span class="op">=</span> <span class="st">"qval"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 2023-05-07 01:32:23.402811 plotOverview: assigning chromosome lengths from hg19!</span></span></code></pre>
<pre><code><span><span class="co">## Scale for <span style="color: #00BB00;">x</span> is already present.</span></span>
<span><span class="co">## Adding another scale for <span style="color: #00BB00;">x</span>, which will replace the existing scale.</span></span>
<span><span class="co">## Scale for <span style="color: #00BB00;">x</span> is already present.</span></span>
<span><span class="co">## Adding another scale for <span style="color: #00BB00;">x</span>, which will replace the existing scale.</span></span></code></pre>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotOverview-1.png" alt="Location of the DERs in the genome. This plot is was designed for many chromosomes but only one is shown here for simplicity." width="700"><p class="caption">
Location of the DERs in the genome. This plot is was designed for many
chromosomes but only one is shown here for simplicity.
</p>
</div>
<p>Figure @ref(fig:plotOverview2) shows the candidate DERs colored by
the type of gene feature they are nearest too.</p>
<div class="sourceCode" id="cb120"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Annotation overview</span></span>
<span><span class="fu"><a href="../reference/plotOverview.html">plotOverview</a></span><span class="op">(</span></span>
<span>    regions <span class="op">=</span> <span class="va">regions</span>, annotation <span class="op">=</span> <span class="va">results</span><span class="op">$</span><span class="va">annotation</span>,</span>
<span>    type <span class="op">=</span> <span class="st">"annotation"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 2023-05-07 01:32:24.808331 plotOverview: assigning chromosome lengths from hg19!</span></span></code></pre>
<pre><code><span><span class="co">## Scale for <span style="color: #00BB00;">x</span> is already present.</span></span>
<span><span class="co">## Adding another scale for <span style="color: #00BB00;">x</span>, which will replace the existing scale.</span></span></code></pre>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotOverview2-1.png" alt="Location of the DERs in the genome and colored by annotation class. This plot is was designed for many chromosomes but only one is shown here for simplicity." width="700"><p class="caption">
Location of the DERs in the genome and colored by annotation class. This
plot is was designed for many chromosomes but only one is shown here for
simplicity.
</p>
</div>
<p>In this particular example, because we are only using data from one
chromosome the above plot is not as informative as in a real case
scenario. However, with this plot we can quickly observe that nearly all
of the candidate DERs are inside an exon.</p>
</div>
<div class="section level3">
<h3 id="plotregioncoverage">
<code>plotRegionCoverage()</code><a class="anchor" aria-label="anchor" href="#plotregioncoverage"></a>
</h3>
<p>The complete opposite of visualizing the candidate DERs at the
genome-level is to visualize them one region at a time.
<code><a href="../reference/plotRegionCoverage.html">plotRegionCoverage()</a></code> allows us to do this quickly for a
large number of regions.</p>
<p>Before using this function, we need to process more detailed
information using two <em><a href="https://bioconductor.org/packages/3.17/derfinder" class="external-link">derfinder</a></em>
functions: <code><a href="https://rdrr.io/pkg/derfinder/man/annotateRegions.html" class="external-link">annotateRegions()</a></code> and
<code><a href="https://rdrr.io/pkg/derfinder/man/getRegionCoverage.html" class="external-link">getRegionCoverage()</a></code> as shown below.</p>
<div class="sourceCode" id="cb123"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Get required information for the plots</span></span>
<span><span class="va">annoRegs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/annotateRegions.html" class="external-link">annotateRegions</a></span><span class="op">(</span><span class="va">regions</span>, <span class="va">genomicState</span><span class="op">$</span><span class="va">fullGenome</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 2023-05-07 01:32:25.527582 annotateRegions: counting</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:32:25.606944 annotateRegions: annotating</span></span></code></pre>
<div class="sourceCode" id="cb126"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">regionCov</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/getRegionCoverage.html" class="external-link">getRegionCoverage</a></span><span class="op">(</span><span class="va">fullCov</span>, <span class="va">regions</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 2023-05-07 01:32:25.725897 getRegionCoverage: processing chr21</span></span></code></pre>
<pre><code><span><span class="co">## 2023-05-07 01:32:25.777701 getRegionCoverage: done processing chr21</span></span></code></pre>
<p>Once we have the relevant information we can proceed to plotting the
first 10 regions. In this case, we will supply
<code><a href="../reference/plotRegionCoverage.html">plotRegionCoverage()</a></code> with the information it needs to plot
transcripts overlapping these 10 regions (Figures @ref(fig:plotRegCov1),
@ref(fig:plotRegCov2), @ref(fig:plotRegCov3), @ref(fig:plotRegCov4),
@ref(fig:plotRegCov5), @ref(fig:plotRegCov6), @ref(fig:plotRegCov7),
@ref(fig:plotRegCov8), @ref(fig:plotRegCov9),
@ref(fig:plotRegCov10)).</p>
<div class="sourceCode" id="cb129"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Plot top 10 regions</span></span>
<span><span class="fu"><a href="../reference/plotRegionCoverage.html">plotRegionCoverage</a></span><span class="op">(</span></span>
<span>    regions <span class="op">=</span> <span class="va">regions</span>, regionCoverage <span class="op">=</span> <span class="va">regionCov</span>,</span>
<span>    groupInfo <span class="op">=</span> <span class="va">pheno</span><span class="op">$</span><span class="va">group</span>, nearestAnnotation <span class="op">=</span> <span class="va">results</span><span class="op">$</span><span class="va">annotation</span>,</span>
<span>    annotatedRegions <span class="op">=</span> <span class="va">annoRegs</span>, whichRegions <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, txdb <span class="op">=</span> <span class="va">txdb</span>, scalefac <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    ask <span class="op">=</span> <span class="cn">FALSE</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotRegCov-1.png" alt="Base-pair resolution plot of differentially expressed region 1." width="700"><p class="caption">
Base-pair resolution plot of differentially expressed region 1.
</p>
</div>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotRegCov-2.png" alt="Base-pair resolution plot of differentially expressed region 2." width="700"><p class="caption">
Base-pair resolution plot of differentially expressed region 2.
</p>
</div>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotRegCov-3.png" alt="Base-pair resolution plot of differentially expressed region 3." width="700"><p class="caption">
Base-pair resolution plot of differentially expressed region 3.
</p>
</div>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotRegCov-4.png" alt="Base-pair resolution plot of differentially expressed region 4." width="700"><p class="caption">
Base-pair resolution plot of differentially expressed region 4.
</p>
</div>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotRegCov-5.png" alt="Base-pair resolution plot of differentially expressed region 5." width="700"><p class="caption">
Base-pair resolution plot of differentially expressed region 5.
</p>
</div>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotRegCov-6.png" alt="Base-pair resolution plot of differentially expressed region 6." width="700"><p class="caption">
Base-pair resolution plot of differentially expressed region 6.
</p>
</div>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotRegCov-7.png" alt="Base-pair resolution plot of differentially expressed region 7." width="700"><p class="caption">
Base-pair resolution plot of differentially expressed region 7.
</p>
</div>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotRegCov-8.png" alt="Base-pair resolution plot of differentially expressed region 8." width="700"><p class="caption">
Base-pair resolution plot of differentially expressed region 8.
</p>
</div>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotRegCov-9.png" alt="Base-pair resolution plot of differentially expressed region 9." width="700"><p class="caption">
Base-pair resolution plot of differentially expressed region 9.
</p>
</div>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotRegCov-10.png" alt="Base-pair resolution plot of differentially expressed region 10." width="700"><p class="caption">
Base-pair resolution plot of differentially expressed region 10.
</p>
</div>
<p>The base-level coverage is shown in a log2 scale with any overlapping
exons shown in dark blue and known introns in light blue.</p>
</div>
<div class="section level3">
<h3 id="plotcluster">
<code>plotCluster()</code><a class="anchor" aria-label="anchor" href="#plotcluster"></a>
</h3>
<p>In this example, we noticed with the
<code><a href="../reference/plotRegionCoverage.html">plotRegionCoverage()</a></code> plots that most of the candidate DERs
are contained in known exons. Sometimes, the signal might be low or we
might have used very stringent cutoffs in the <em><a href="https://bioconductor.org/packages/3.17/derfinder" class="external-link">derfinder</a></em>
analysis. One way we can observe this is by plotting clusters of regions
where a cluster is defined as regions within 300 bp (default option) of
each other.</p>
<p>To visualize the clusters, we can use <code><a href="../reference/plotCluster.html">plotCluster()</a></code>
which takes similar input to <code><a href="../reference/plotOverview.html">plotOverview()</a></code> with the
notable addition of the coverage information as well as the
<code>idx</code> argument. This argument specifies which region to focus
on: it will be plotted with a red bar and will determine the cluster to
display.</p>
<p>In Figure @ref(fig:plotCluster) we observe one large candidate DER
with other nearby ones that do not have a q-value less than 0.10. In a
real analysis, we would probably discard this region as the coverage is
fairly low.</p>
<div class="sourceCode" id="cb130"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## First cluster</span></span>
<span><span class="fu"><a href="../reference/plotCluster.html">plotCluster</a></span><span class="op">(</span></span>
<span>    idx <span class="op">=</span> <span class="fl">1</span>, regions <span class="op">=</span> <span class="va">regions</span>, annotation <span class="op">=</span> <span class="va">results</span><span class="op">$</span><span class="va">annotation</span>,</span>
<span>    coverageInfo <span class="op">=</span> <span class="va">fullCov</span><span class="op">$</span><span class="va">chr21</span>, txdb <span class="op">=</span> <span class="va">txdb</span>, groupInfo <span class="op">=</span> <span class="va">pheno</span><span class="op">$</span><span class="va">group</span>,</span>
<span>    titleUse <span class="op">=</span> <span class="st">"pval"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Parsing transcripts...</span></span></code></pre>
<pre><code><span><span class="co">## Parsing exons...</span></span></code></pre>
<pre><code><span><span class="co">## Parsing cds...</span></span></code></pre>
<pre><code><span><span class="co">## Parsing utrs...</span></span></code></pre>
<pre><code><span><span class="co">## ------exons...</span></span></code></pre>
<pre><code><span><span class="co">## ------cdss...</span></span></code></pre>
<pre><code><span><span class="co">## ------introns...</span></span></code></pre>
<pre><code><span><span class="co">## ------utr...</span></span></code></pre>
<pre><code><span><span class="co">## aggregating...</span></span></code></pre>
<pre><code><span><span class="co">## Done</span></span></code></pre>
<pre><code><span><span class="co">## Constructing graphics...</span></span></code></pre>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotCluster-1.png" alt="Cluster plot for cluster 1 using ggbio." width="700"><p class="caption">
Cluster plot for cluster 1 using ggbio.
</p>
</div>
<p>The second cluster (Figure @ref(fig:plotCluster2)) shows a larger
number of potential DERs (again without q-values less than 0.10) in a
segment of the genome where the coverage data is highly variable. This
is a common occurrence with RNA-seq data.</p>
<div class="sourceCode" id="cb142"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Second cluster</span></span>
<span><span class="fu"><a href="../reference/plotCluster.html">plotCluster</a></span><span class="op">(</span></span>
<span>    idx <span class="op">=</span> <span class="fl">2</span>, regions <span class="op">=</span> <span class="va">regions</span>, annotation <span class="op">=</span> <span class="va">results</span><span class="op">$</span><span class="va">annotation</span>,</span>
<span>    coverageInfo <span class="op">=</span> <span class="va">fullCov</span><span class="op">$</span><span class="va">chr21</span>, txdb <span class="op">=</span> <span class="va">txdb</span>, groupInfo <span class="op">=</span> <span class="va">pheno</span><span class="op">$</span><span class="va">group</span>,</span>
<span>    titleUse <span class="op">=</span> <span class="st">"pval"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Parsing transcripts...</span></span></code></pre>
<pre><code><span><span class="co">## Parsing exons...</span></span></code></pre>
<pre><code><span><span class="co">## Parsing cds...</span></span></code></pre>
<pre><code><span><span class="co">## Parsing utrs...</span></span></code></pre>
<pre><code><span><span class="co">## ------exons...</span></span></code></pre>
<pre><code><span><span class="co">## ------cdss...</span></span></code></pre>
<pre><code><span><span class="co">## ------introns...</span></span></code></pre>
<pre><code><span><span class="co">## ------utr...</span></span></code></pre>
<pre><code><span><span class="co">## aggregating...</span></span></code></pre>
<pre><code><span><span class="co">## Done</span></span></code></pre>
<pre><code><span><span class="co">## Constructing graphics...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in !vapply(ggl, fixed, logical(1L)) &amp; !vapply(PlotList, is, "Ideogram",</span></span>
<span><span class="co">## : longer object length is not a multiple of shorter object length</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Transformation introduced infinite values in continuous y-axis</span></span></code></pre>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotCluster2-1.png" alt="Cluster plot for cluster 2 using ggbio." width="700"><p class="caption">
Cluster plot for cluster 2 using ggbio.
</p>
</div>
<p>These plots show an ideogram which helps quickly identify which
region of the genome we are focusing on. Then, the base-level coverage
information for each sample is displayed in log2. Next, the coverage
group means are shown in the log2 scale. The plot is completed with the
potential and candidate DERs as well as any known transcripts.</p>
</div>
<div class="section level3">
<h3 id="vennregions">
<code>vennRegions</code><a class="anchor" aria-label="anchor" href="#vennregions"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.17/derfinder" class="external-link">derfinder</a></em>
has functions for annotating regions given their genomic state. A
typical visualization is to then view how many regions overlap known
exons, introns, intergenic regions, none of them or several of these
groups in a venn diagram. The function <code><a href="../reference/vennRegions.html">vennRegions()</a></code> makes
this plot using the output from
<code><a href="https://rdrr.io/pkg/derfinder/man/annotateRegions.html" class="external-link">derfinder::annotateRegions()</a></code> as shown in Figure
@ref(fig:vennRegions).</p>
<div class="sourceCode" id="cb156"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Make venn diagram</span></span>
<span><span class="va">venn</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/vennRegions.html">vennRegions</a></span><span class="op">(</span><span class="va">annoRegs</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="derfinderPlot_files/figure-html/vennRegions-1.png" alt="Venn diagram of regions by annotation class." width="700"><p class="caption">
Venn diagram of regions by annotation class.
</p>
</div>
<div class="sourceCode" id="cb157"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## It returns the actual venn counts information</span></span>
<span><span class="va">venn</span></span></code></pre></div>
<pre><code><span><span class="co">##   exon intergenic intron Counts</span></span>
<span><span class="co">## 1    0          0      0      0</span></span>
<span><span class="co">## 2    0          0      1      2</span></span>
<span><span class="co">## 3    0          1      0      4</span></span>
<span><span class="co">## 4    0          1      1      0</span></span>
<span><span class="co">## 5    1          0      0    259</span></span>
<span><span class="co">## 6    1          0      1     35</span></span>
<span><span class="co">## 7    1          1      0      0</span></span>
<span><span class="co">## 8    1          1      1      0</span></span>
<span><span class="co">## attr(,"class")</span></span>
<span><span class="co">## [1] "VennCounts"</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>This package was made possible thanks to:</p>
<ul>
<li>R (R Core Team, 2023)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/GenomeInfoDb" class="external-link">GenomeInfoDb</a></em>
(Arora, Morgan, Carlson, and Pagès, 2017)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/GenomicRanges" class="external-link">GenomicRanges</a></em>
(Lawrence, Huber, Pagès, Aboyoun, Carlson, Gentleman, Morgan, and Carey,
2013)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/ggbio" class="external-link">ggbio</a></em> (Yin,
Cook, and Lawrence, 2012)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=ggplot2" class="external-link">ggplot2</a></em>
(Wickham, 2016)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/IRanges" class="external-link">IRanges</a></em>
(Lawrence, Huber, Pagès et al., 2013)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=plyr" class="external-link">plyr</a></em>
(Wickham, 2011)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RColorBrewer" class="external-link">RColorBrewer</a></em>
(Neuwirth, 2022)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=reshape2" class="external-link">reshape2</a></em>
(Wickham, 2007)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=scales" class="external-link">scales</a></em>
(Wickham and Seidel, 2022)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/biovizBase" class="external-link">biovizBase</a></em>
(Yin, Lawrence, and Cook, 2023)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/bumphunter" class="external-link">bumphunter</a></em>
(Jaffe, Murakami, Lee, Leek, Fallin, Feinberg, and Irizarry, 2012) and
(Jaffe, Murakami, Lee, Leek, Fallin, Feinberg, and Irizarry, 2012)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/derfinder" class="external-link">derfinder</a></em>
(Collado-Torres, Nellore, Frazee et al., 2017)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/derfinderData" class="external-link">derfinderData</a></em>
(Collado-Torres, Jaffe, and Leek, 2023)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight, Müller, and Hester, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2014)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2023)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux, McPherson, Luraschi, Ushey, Atkins, Wickham,
Cheng, Chang, and Iannone, 2023)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/TxDb.Hsapiens.UCSC.hg19.knownGene" class="external-link">TxDb.Hsapiens.UCSC.hg19.knownGene</a></em>
(Carlson and Maintainer, 2015)</li>
</ul>
<p>Code for creating the vignette</p>
<div class="sourceCode" id="cb159"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Create the vignette</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/rstudio/rmarkdown" class="external-link">"rmarkdown"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="fu"><a href="https://pkgs.rstudio.com/rmarkdown/reference/render.html" class="external-link">render</a></span><span class="op">(</span><span class="st">"derfinderPlot.Rmd"</span>, <span class="st">"BiocStyle::html_document"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Extract the R code</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://yihui.org/knitr/" class="external-link">"knitr"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/knit.html" class="external-link">knit</a></span><span class="op">(</span><span class="st">"derfinderPlot.Rmd"</span>, tangle <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb160"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Clean up</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="st">"chr21"</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Date the vignette was generated.</p>
<pre><code><span><span class="co">## [1] "2023-05-07 01:32:49 UTC"</span></span></code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code><span><span class="co">## Time difference of 1.666 mins</span></span></code></pre>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">## <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">##  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">##  version  R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">##  os       Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">##  system   x86_64, linux-gnu</span></span>
<span><span class="co">##  ui       X11</span></span>
<span><span class="co">##  language en</span></span>
<span><span class="co">##  collate  en_US.UTF-8</span></span>
<span><span class="co">##  ctype    en_US.UTF-8</span></span>
<span><span class="co">##  tz       UTC</span></span>
<span><span class="co">##  date     2023-05-07</span></span>
<span><span class="co">##  pandoc   2.19.2 @ /usr/local/bin/ (via rmarkdown)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">##  <span style="color: #555555; font-style: italic;">package                          </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version  </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">##  AnnotationDbi                     * 1.62.1    <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  AnnotationFilter                    1.24.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  backports                           1.4.1     <span style="color: #555555;">2021-12-13</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  base64enc                           0.1-3     <span style="color: #555555;">2015-07-28</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  bibtex                              0.5.1     <span style="color: #555555;">2023-01-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  Biobase                           * 2.60.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocFileCache                       2.8.0     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocGenerics                      * 0.46.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocIO                              1.10.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocManager                         1.30.20   <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  BiocParallel                        1.34.1    <span style="color: #555555;">2023-05-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  BiocStyle                         * 2.28.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  biomaRt                             2.56.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  Biostrings                          2.68.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  biovizBase                          1.48.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  bit                                 4.0.5     <span style="color: #555555;">2022-11-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  bit64                               4.0.5     <span style="color: #555555;">2020-08-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  bitops                              1.0-7     <span style="color: #555555;">2021-04-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  blob                                1.2.4     <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  bookdown                            0.33      <span style="color: #555555;">2023-03-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  BSgenome                            1.68.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  bslib                               0.4.2     <span style="color: #555555;">2022-12-16</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  bumphunter                        * 1.42.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  cachem                              1.0.8     <span style="color: #555555;">2023-05-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  checkmate                           2.2.0     <span style="color: #555555;">2023-04-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  cli                                 3.6.1     <span style="color: #555555;">2023-03-23</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  cluster                             2.1.4     <span style="color: #555555;">2022-08-22</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  codetools                           0.2-19    <span style="color: #555555;">2023-02-01</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  colorspace                          2.1-0     <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  crayon                              1.5.2     <span style="color: #555555;">2022-09-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  curl                                5.0.0     <span style="color: #555555;">2023-01-12</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  data.table                          1.14.8    <span style="color: #555555;">2023-02-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  DBI                                 1.1.3     <span style="color: #555555;">2022-06-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  dbplyr                              2.3.2     <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  DelayedArray                        0.26.2    <span style="color: #555555;">2023-05-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  derfinder                         * 1.34.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  derfinderData                     * 2.18.0    <span style="color: #555555;">2023-04-27</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  derfinderHelper                     1.34.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  derfinderPlot                     * 1.35.0    <span style="color: #555555;">2023-05-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  desc                                1.4.2     <span style="color: #555555;">2022-09-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  dichromat                           2.0-0.1   <span style="color: #555555;">2022-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  digest                              0.6.31    <span style="color: #555555;">2022-12-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  doRNG                               1.8.6     <span style="color: #555555;">2023-01-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  dplyr                               1.1.2     <span style="color: #555555;">2023-04-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  ensembldb                           2.24.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  evaluate                            0.20      <span style="color: #555555;">2023-01-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  fansi                               1.0.4     <span style="color: #555555;">2023-01-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  farver                              2.1.1     <span style="color: #555555;">2022-07-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  fastmap                             1.1.1     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  filelock                            1.0.2     <span style="color: #555555;">2018-10-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  foreach                           * 1.5.2     <span style="color: #555555;">2022-02-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  foreign                             0.8-84    <span style="color: #555555;">2022-12-06</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  Formula                             1.2-5     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  fs                                  1.6.2     <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  generics                            0.1.3     <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  GenomeInfoDb                      * 1.36.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GenomeInfoDbData                    1.2.10    <span style="color: #555555;">2023-05-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GenomicAlignments                   1.36.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GenomicFeatures                   * 1.52.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GenomicFiles                        1.36.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GenomicRanges                     * 1.52.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  GGally                              2.1.2     <span style="color: #555555;">2021-06-21</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  ggbio                               1.48.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  ggplot2                             3.4.2     <span style="color: #555555;">2023-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  glue                                1.6.2     <span style="color: #555555;">2022-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  graph                               1.78.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  gridExtra                           2.3       <span style="color: #555555;">2017-09-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  gtable                              0.3.3     <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  highr                               0.10      <span style="color: #555555;">2022-12-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  Hmisc                               5.0-1     <span style="color: #555555;">2023-03-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  hms                                 1.1.3     <span style="color: #555555;">2023-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  htmlTable                           2.4.1     <span style="color: #555555;">2022-07-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  htmltools                           0.5.5     <span style="color: #555555;">2023-03-23</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  htmlwidgets                         1.6.2     <span style="color: #555555;">2023-03-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  httr                                1.4.5     <span style="color: #555555;">2023-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  IRanges                           * 2.34.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  iterators                         * 1.0.14    <span style="color: #555555;">2022-02-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  jquerylib                           0.1.4     <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  jsonlite                            1.8.4     <span style="color: #555555;">2022-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  KEGGREST                            1.40.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  knitr                             * 1.42      <span style="color: #555555;">2023-01-25</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  labeling                            0.4.2     <span style="color: #555555;">2020-10-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  lattice                             0.21-8    <span style="color: #555555;">2023-04-05</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  lazyeval                            0.2.2     <span style="color: #555555;">2019-03-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  lifecycle                           1.0.3     <span style="color: #555555;">2022-10-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  limma                               3.56.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  locfit                            * 1.5-9.7   <span style="color: #555555;">2023-01-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  lubridate                           1.9.2     <span style="color: #555555;">2023-02-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  magrittr                            2.0.3     <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  Matrix                              1.5-4     <span style="color: #555555;">2023-04-04</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  MatrixGenerics                      1.12.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  matrixStats                         0.63.0    <span style="color: #555555;">2022-11-18</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  memoise                             2.0.1     <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  munsell                             0.5.0     <span style="color: #555555;">2018-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  nnet                                7.3-19    <span style="color: #555555;">2023-05-03</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  org.Hs.eg.db                      * 3.17.0    <span style="color: #555555;">2023-05-07</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  OrganismDbi                         1.42.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  pillar                              1.9.0     <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  pkgconfig                           2.0.3     <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  pkgdown                             2.0.7     <span style="color: #555555;">2022-12-14</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  plyr                                1.8.8     <span style="color: #555555;">2022-11-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  png                                 0.1-8     <span style="color: #555555;">2022-11-29</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  prettyunits                         1.1.1     <span style="color: #555555;">2020-01-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  progress                            1.2.2     <span style="color: #555555;">2019-05-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  ProtGenerics                        1.32.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  purrr                               1.0.1     <span style="color: #555555;">2023-01-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  qvalue                              2.32.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  R6                                  2.5.1     <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  ragg                                1.2.5     <span style="color: #555555;">2023-01-12</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  rappdirs                            0.3.3     <span style="color: #555555;">2021-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  RBGL                                1.76.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  RColorBrewer                        1.1-3     <span style="color: #555555;">2022-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  Rcpp                                1.0.10    <span style="color: #555555;">2023-01-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  RCurl                               1.98-1.12 <span style="color: #555555;">2023-03-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  RefManageR                        * 1.4.0     <span style="color: #555555;">2022-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  reshape                             0.8.9     <span style="color: #555555;">2022-04-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  reshape2                            1.4.4     <span style="color: #555555;">2020-04-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  restfulr                            0.0.15    <span style="color: #555555;">2022-06-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  rjson                               0.2.21    <span style="color: #555555;">2022-01-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  rlang                               1.1.1     <span style="color: #555555;">2023-04-28</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  rmarkdown                           2.21      <span style="color: #555555;">2023-03-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  rngtools                            1.5.2     <span style="color: #555555;">2021-09-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  rpart                               4.1.19    <span style="color: #555555;">2022-10-21</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  rprojroot                           2.0.3     <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  Rsamtools                           2.16.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  RSQLite                             2.3.1     <span style="color: #555555;">2023-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  rstudioapi                          0.14      <span style="color: #555555;">2022-08-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  rtracklayer                         1.60.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  S4Arrays                            1.0.1     <span style="color: #555555;">2023-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  S4Vectors                         * 0.38.1    <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  sass                                0.4.6     <span style="color: #555555;">2023-05-03</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  scales                              1.2.1     <span style="color: #555555;">2022-08-20</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  sessioninfo                       * 1.2.2     <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  stringi                             1.7.12    <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  stringr                             1.5.0     <span style="color: #555555;">2022-12-02</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  SummarizedExperiment                1.30.1    <span style="color: #555555;">2023-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  systemfonts                         1.0.4     <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  textshaping                         0.3.6     <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  tibble                              3.2.1     <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  tidyselect                          1.2.0     <span style="color: #555555;">2022-10-10</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">CRAN (R 4.3.0)</span></span></span>
<span><span class="co">##  timechange                          0.2.0     <span style="color: #555555;">2023-01-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  TxDb.Hsapiens.UCSC.hg19.knownGene * 3.2.2     <span style="color: #555555;">2022-12-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  utf8                                1.2.3     <span style="color: #555555;">2023-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  VariantAnnotation                   1.46.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  vctrs                               0.6.2     <span style="color: #555555;">2023-04-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  withr                               2.5.0     <span style="color: #555555;">2022-03-03</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  xfun                                0.39      <span style="color: #555555;">2023-04-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  XML                                 3.99-0.14 <span style="color: #555555;">2023-03-19</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  xml2                                1.3.4     <span style="color: #555555;">2023-04-27</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  XVector                             0.40.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">##  yaml                                2.3.7     <span style="color: #555555;">2023-01-23</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.3.0)</span></span></span>
<span><span class="co">##  zlibbioc                            1.46.0    <span style="color: #555555;">2023-04-25</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #555555;"> [1] /__w/_temp/Library</span></span></span>
<span><span class="co">## <span style="color: #555555;"> [2] /usr/local/lib/R/site-library</span></span></span>
<span><span class="co">## <span style="color: #555555;"> [3] /usr/local/lib/R/library</span></span></span>
<span><span class="co">## </span></span>
<span><span class="co">## <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.17/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2023) with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2014) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2023) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2023rmarkdown"></a><a href="#cite-allaire2023rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.21. 2023. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-GenomeInfoDb"></a><a href="#cite-GenomeInfoDb">[2]</a><cite>
S. Arora, M. Morgan, M. Carlson, et al. <em>GenomeInfoDb: Utilities for
manipulating chromosome and other ‘seqname’ identifiers</em>. 2017. DOI:
<a href="https://doi.org/10.18129/B9.bioc.GenomeInfoDb" class="external-link">10.18129/B9.bioc.GenomeInfoDb</a>.</cite>
</p>
<p>
<a id="bib-brainspan"></a><a href="#cite-brainspan">[3]</a><cite>
BrainSpan. “Atlas of the Developing Human Brain [Internet]. Funded by
ARRA Awards 1RC2MH089921-01, 1RC2MH090047-01, and 1RC2MH089929-01.”
2011. URL:
<a href="http://www.brainspan.org/" class="external-link">http://www.brainspan.org/</a>.</cite>
</p>
<p>
<a id="bib-carlson2015txdb"></a><a href="#cite-carlson2015txdb">[4]</a><cite>
M. Carlson and B. P. Maintainer. <em>TxDb.Hsapiens.UCSC.hg19.knownGene:
Annotation package for TxDb object(s)</em>. R package version 3.2.2.
2015.</cite>
</p>
<p>
<a id="bib-colladotorres2017derfinderplot"></a><a href="#cite-colladotorres2017derfinderplot">[5]</a><cite>
L. Collado-Torres, A. E. Jaffe, and J. T. Leek. <em>derfinderPlot:
Plotting functions for derfinder</em>. <a href="https://github.com/leekgroup/derfinderPlot" class="external-link uri">https://github.com/leekgroup/derfinderPlot</a> - R package
version 1.35.0. 2017. DOI:
<a href="https://doi.org/10.18129/B9.bioc.derfinderPlot" class="external-link">10.18129/B9.bioc.derfinderPlot</a>.
URL:
<a href="http://www.bioconductor.org/packages/derfinderPlot" class="external-link">http://www.bioconductor.org/packages/derfinderPlot</a>.</cite>
</p>
<p>
<a id="bib-colladotorres2023derfinderdata"></a><a href="#cite-colladotorres2023derfinderdata">[6]</a><cite>
L. Collado-Torres, A. Jaffe, and J. Leek. <em>derfinderData: Processed
BigWigs from BrainSpan for examples</em>. R package version 2.18.0.
2023. DOI:
<a href="https://doi.org/10.18129/B9.bioc.derfinderData" class="external-link">10.18129/B9.bioc.derfinderData</a>.
URL:
<a href="https://bioconductor.org/packages/derfinderData" class="external-link">https://bioconductor.org/packages/derfinderData</a>.</cite>
</p>
<p>
<a id="bib-colladotorres2017flexible"></a><a href="#cite-colladotorres2017flexible">[7]</a><cite>
L. Collado-Torres, A. Nellore, A. C. Frazee, et al. “Flexible expressed
region analysis for RNA-seq with derfinder”. In: <em>Nucl. Acids
Res.</em> (2017). DOI:
<a href="https://doi.org/10.1093/nar/gkw852" class="external-link">10.1093/nar/gkw852</a>.
URL:
<a href="http://nar.oxfordjournals.org/content/early/2016/09/29/nar.gkw852" class="external-link">http://nar.oxfordjournals.org/content/early/2016/09/29/nar.gkw852</a>.</cite>
</p>
<p>
<a id="bib-bumphunter"></a><a href="#cite-bumphunter">[8]</a><cite> A.
E. Jaffe, P. Murakami, H. Lee, et al. “Bump hunting to identify
differentially methylated regions in epigenetic epidemiology studies”.
In: <em>International journal of epidemiology</em> 41.1 (2012),
pp. 200–209. DOI:
<a href="https://doi.org/10.1093/ije/dyr238" class="external-link">10.1093/ije/dyr238</a>.</cite>
</p>
<p>
<a id="bib-bumphunterPaper"></a><a href="#cite-bumphunterPaper">[9]</a><cite>
A. E. Jaffe, P. Murakami, H. Lee, et al. “Bump hunting to identify
differentially methylated regions in epigenetic epidemiology studies”.
In: <em>International Journal of Epidemiology</em> (2012).</cite>
</p>
<p>
<a id="bib-lawrence2013software"></a><a href="#cite-lawrence2013software">[10]</a><cite>
M. Lawrence, W. Huber, H. Pagès, et al. “Software for Computing and
Annotating Genomic Ranges”. In: <em>PLoS Computational Biology</em> 9 (8
2013). DOI:
<a href="https://doi.org/10.1371/journal.pcbi.1003118" class="external-link">10.1371/journal.pcbi.1003118</a>.
URL:
<a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118%7D." class="external-link"></a><a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118" class="external-link uri">http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118</a>}.
</cite></p>


<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[11]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-neuwirth2022rcolorbrewer"></a><a href="#cite-neuwirth2022rcolorbrewer">[12]</a><cite>
E. Neuwirth. <em>RColorBrewer: ColorBrewer Palettes</em>. R package
version 1.1-3. 2022. URL:
<a href="https://CRAN.R-project.org/package=RColorBrewer" class="external-link">https://CRAN.R-project.org/package=RColorBrewer</a>.</cite>
</p>
<p>
<a id="bib-ole2023biocstyle"></a><a href="#cite-ole2023biocstyle">[13]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.28.0. 2023. DOI:
<a href="https://doi.org/10.18129/B9.bioc.BiocStyle" class="external-link">10.18129/B9.bioc.BiocStyle</a>.
URL:
<a href="https://bioconductor.org/packages/BiocStyle" class="external-link">https://bioconductor.org/packages/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2023language"></a><a href="#cite-2023language">[14]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2023. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-wickham2016ggplot2"></a><a href="#cite-wickham2016ggplot2">[15]</a><cite>
H. Wickham. <em>ggplot2: Elegant Graphics for Data Analysis</em>.
Springer-Verlag New York, 2016. ISBN: 978-3-319-24277-4. URL:
<a href="https://ggplot2.tidyverse.org" class="external-link">https://ggplot2.tidyverse.org</a>.</cite>
</p>
<p>
<a id="bib-wickham2007reshaping"></a><a href="#cite-wickham2007reshaping">[16]</a><cite>
H. Wickham. “Reshaping Data with the reshape Package”. In: <em>Journal
of Statistical Software</em> 21.12 (2007), pp. 1–20. URL:
<a href="http://www.jstatsoft.org/v21/i12/" class="external-link">http://www.jstatsoft.org/v21/i12/</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[17]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2011split"></a><a href="#cite-wickham2011split">[18]</a><cite>
H. Wickham. “The Split-Apply-Combine Strategy for Data Analysis”. In:
<em>Journal of Statistical Software</em> 40.1 (2011), pp. 1–29. URL:
<a href="https://www.jstatsoft.org/v40/i01/" class="external-link">https://www.jstatsoft.org/v40/i01/</a>.</cite>
</p>
<p>
<a id="bib-wickham2021sessioninfo"></a><a href="#cite-wickham2021sessioninfo">[19]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. <a href="https://github.com/r-lib/sessioninfo#readme" class="external-link uri">https://github.com/r-lib/sessioninfo#readme</a>, <a href="https://r-lib.github.io/sessioninfo/" class="external-link uri">https://r-lib.github.io/sessioninfo/</a>. 2021.</cite>
</p>
<p>
<a id="bib-wickham2022scales"></a><a href="#cite-wickham2022scales">[20]</a><cite>
H. Wickham and D. Seidel. <em>scales: Scale Functions for
Visualization</em>. R package version 1.2.1. 2022. URL:
<a href="https://CRAN.R-project.org/package=scales" class="external-link">https://CRAN.R-project.org/package=scales</a>.</cite>
</p>
<p>
<a id="bib-xie2014knitr"></a><a href="#cite-xie2014knitr">[21]</a><cite>
Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in R”.
In: <em>Implementing Reproducible Computational Research</em>. Ed. by V.
Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and
Hall/CRC, 2014.</cite>
</p>
<p>
<a id="bib-yin2012ggbio"></a><a href="#cite-yin2012ggbio">[22]</a><cite>
T. Yin, D. Cook, and M. Lawrence. “ggbio: an R package for extending the
grammar of graphics for genomic data”. In: <em>Genome Biology</em> 13.8
(2012), p. R77.</cite>
</p>
<p>
<a id="bib-yin2023biovizbase"></a><a href="#cite-yin2023biovizbase">[23]</a><cite>
T. Yin, M. Lawrence, and D. Cook. <em>biovizBase: Basic graphic
utilities for visualization of genomic data.</em> R package version
1.48.0. 2023. DOI:
<a href="https://doi.org/10.18129/B9.bioc.biovizBase" class="external-link">10.18129/B9.bioc.biovizBase</a>.
URL:
<a href="https://bioconductor.org/packages/biovizBase" class="external-link">https://bioconductor.org/packages/biovizBase</a>.</cite>
</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Leonardo Collado-Torres, Andrew E. Jaffe, Jeffrey T. Leek.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
