<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to derfinderPlot • derfinderPlot</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to derfinderPlot">
<meta property="og:description" content="derfinderPlot">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">derfinderPlot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.21.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/derfinderPlot.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/leekgroup/derfinderPlot/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to derfinderPlot</h1>
                        <h4 class="author">Leonardo Collado-Torres</h4>
            <address class="author_afil">
      Lieber Institute for Brain Development, Johns Hopkins Medical Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:lcolladotor@gmail.com" class="email">lcolladotor@gmail.com</a>
      </address>
                  
            <h4 class="date">27 April 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/leekgroup/derfinderPlot/blob/master/vignettes/derfinderPlot.Rmd"><code>vignettes/derfinderPlot.Rmd</code></a></small>
      <div class="hidden name"><code>derfinderPlot.Rmd</code></div>

    </div>

    
    
<div id="basics" class="section level1">
<h1 class="hasAnchor">
<a href="#basics" class="anchor"></a>Basics</h1>
<div id="install-derfinderplot" class="section level2">
<h2 class="hasAnchor">
<a href="#install-derfinderplot" class="anchor"></a>Install <em><a href="https://bioconductor.org/packages/3.11/derfinderPlot">derfinderPlot</a></em>
</h2>
<p><code>R</code> is an open-source statistical environment which can be easily modified to enhance its functionality via packages. <em><a href="https://bioconductor.org/packages/3.11/derfinderPlot">derfinderPlot</a></em> is a <code>R</code> package available via the <a href="http://bioconductor/packages/derfinderPlot">Bioconductor</a> repository for packages. <code>R</code> can be installed on any operating system from <a href="https://cran.r-project.org/">CRAN</a> after which you can install <em><a href="https://bioconductor.org/packages/3.11/derfinderPlot">derfinderPlot</a></em> by using the following commands in your <code>R</code> session:</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="kw">if</span> (!<span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"BiocManager"</span>, <span class="kw">quietly</span> <span class="kw">=</span> <span class="fl">TRUE</span>))
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>)

<span class="kw pkg">BiocManager</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"derfinderPlot"</span>)

<span class="co">## Check that you have a valid Bioconductor installation</span>
<span class="kw pkg">BiocManager</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/valid.html">valid</a></span>()</pre></body></html></div>
</div>
<div id="required-knowledge" class="section level2">
<h2 class="hasAnchor">
<a href="#required-knowledge" class="anchor"></a>Required knowledge</h2>
<p><em><a href="https://bioconductor.org/packages/3.11/derfinderPlot">derfinderPlot</a></em> is based on many other packages and in particular in those that have implemented the infrastructure needed for dealing with RNA-seq data. A <em><a href="https://bioconductor.org/packages/3.11/derfinderPlot">derfinderPlot</a></em> user is not expected to deal with those packages directly but will need to be familiar with <em><a href="https://bioconductor.org/packages/3.11/derfinder">derfinder</a></em> and for some plots with <em><a href="https://bioconductor.org/packages/3.11/ggbio">ggbio</a></em>.</p>
<p>If you are asking yourself the question “Where do I start using Bioconductor?” you might be interested in <a href="http://lcolladotor.github.io/2014/10/16/startbioc/#.VkOKbq6rRuU">this blog post</a>.</p>
</div>
<div id="asking-for-help" class="section level2">
<h2 class="hasAnchor">
<a href="#asking-for-help" class="anchor"></a>Asking for help</h2>
<p>As package developers, we try to explain clearly how to use our packages and in which order to use the functions. But <code>R</code> and <code>Bioconductor</code> have a steep learning curve so it is critical to learn where to ask for help. The blog post quoted above mentions some but we would like to highlight the <a href="https://support.bioconductor.org/">Bioconductor support site</a> as the main resource for getting help: remember to use the <code>derfinder</code> or <code>derfinderPlot</code> tags and check <a href="https://support.bioconductor.org/t/derfinder/">the older posts</a>. Other alternatives are available such as creating GitHub issues and tweeting. However, please note that if you want to receive help you should adhere to the <a href="http://www.bioconductor.org/help/support/posting-guide/">posting guidelines</a>. It is particularly critical that you provide a small reproducible example and your session information so package developers can track down the source of the error.</p>
</div>
<div id="citing-derfinderplot" class="section level2">
<h2 class="hasAnchor">
<a href="#citing-derfinderplot" class="anchor"></a>Citing <em><a href="https://bioconductor.org/packages/3.11/derfinderPlot">derfinderPlot</a></em>
</h2>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.11/derfinderPlot">derfinderPlot</a></em> will be useful for your research. Please use the following information to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="co">## Citation info</span>
<span class="fu"><a href="https://rdrr.io/r/utils/citation.html">citation</a></span>(<span class="st">"derfinderPlot"</span>)</pre></body></html></div>
<pre><code>## 
## Collado-Torres L, Jaffe AE, Leek JT (2017). _derfinderPlot: Plotting
## functions for derfinder_. doi: 10.18129/B9.bioc.derfinderPlot (URL:
## https://doi.org/10.18129/B9.bioc.derfinderPlot),
## https://github.com/leekgroup/derfinderPlot - R package version 1.21.5,
## &lt;URL: http://www.bioconductor.org/packages/derfinderPlot&gt;.
## 
## Collado-Torres L, Nellore A, Frazee AC, Wilks C, Love MI, Langmead B,
## Irizarry RA, Leek JT, Jaffe AE (2017). "Flexible expressed region
## analysis for RNA-seq with derfinder." _Nucl. Acids Res._. doi:
## 10.1093/nar/gkw852 (URL: https://doi.org/10.1093/nar/gkw852), &lt;URL:
## http://nar.oxfordjournals.org/content/early/2016/09/29/nar.gkw852&gt;.
## 
## To see these entries in BibTeX format, use 'print(&lt;citation&gt;,
## bibtex=TRUE)', 'toBibtex(.)', or set
## 'options(citation.bibtex.max=999)'.</code></pre>
</div>
</div>
<div id="introduction-to-derfinderplot" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction-to-derfinderplot" class="anchor"></a>Introduction to <em><a href="https://bioconductor.org/packages/3.11/derfinderPlot">derfinderPlot</a></em>
</h1>
<p><em><a href="https://bioconductor.org/packages/3.11/derfinderPlot">derfinderPlot</a></em> <a id="cite-Collado-Torres_2017"></a>(<a href="http://www.bioconductor.org/packages/derfinderPlot">Collado-Torres, Jaffe, and Leek, 2017</a>) is an addon package for <em><a href="https://bioconductor.org/packages/3.11/derfinder">derfinder</a></em> <a id="cite-Collado-Torres_2017a"></a>(<a href="http://nar.oxfordjournals.org/content/early/2016/09/29/nar.gkw852">Collado-Torres, Nellore, Frazee, Wilks, et al., 2017</a>) with functions that allow you to visualize the results.</p>
<p>While the functions in <em><a href="https://bioconductor.org/packages/3.11/derfinderPlot">derfinderPlot</a></em> assume you generated the data with <em><a href="https://bioconductor.org/packages/3.11/derfinder">derfinder</a></em>, they can be used with other <code>GRanges</code> objects properly formatted.</p>
<p>The functions in <em><a href="https://bioconductor.org/packages/3.11/derfinderPlot">derfinderPlot</a></em> are:</p>
<ul>
<li>
<code><a href="../reference/plotCluster.html">plotCluster()</a></code> is a tailored <em><a href="https://bioconductor.org/packages/3.11/ggbio">ggbio</a></em> <a id="cite-Yin_2012"></a>(<a href="#bib-Yin_2012">Yin, Cook, and Lawrence, 2012</a>) plot that shows all the regions in a cluster (defined by distance). It shows the base-level coverage for each sample as well as the mean for each group. If these regions overlap any known gene, the gene and the transcript annotation is displayed.</li>
<li>
<code><a href="../reference/plotOverview.html">plotOverview()</a></code> is another tailored <em><a href="https://bioconductor.org/packages/3.11/ggbio">ggbio</a></em> (<a href="#bib-Yin_2012">Yin, Cook, and Lawrence, 2012</a>) plot showing an overview of the whole genome. This plot can be useful to observe if the regions are clustered in a subset of a chromosome. It can also be used to check whether the regions match predominantly one part of the gene structure (for example, 3’ overlaps).</li>
<li>
<code><a href="../reference/plotRegionCoverage.html">plotRegionCoverage()</a></code> is a fast plotting function using <code>R</code> base graphics that shows the base-level coverage for each sample inside a specific region of the genome. If the region overlaps any known gene or intron, the information is displayed. Optionally, it can display the known transcripts. This function is most likely the easiest to use with <code>GRanges</code> objects from other packages.</li>
</ul>
</div>
<div id="example" class="section level1">
<h1 class="hasAnchor">
<a href="#example" class="anchor"></a>Example</h1>
<p>As an example, we will analyze a small subset of the samples from the <em>BrainSpan Atlas of the Human Brain</em> <a id="cite-brainspan"></a>(<a href="http://www.brainspan.org/">BrainSpan, 2011</a>) publicly available data.</p>
<p>We first load the required packages.</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="co">## Load libraries</span>
<span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"derfinder"</span>))
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"derfinderData"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"derfinderPlot"</span>)</pre></body></html></div>
<pre><code>## Registered S3 method overwritten by 'GGally':
##   method from   
##   +.gg   ggplot2</code></pre>
<div id="analyze-data" class="section level2">
<h2 class="hasAnchor">
<a href="#analyze-data" class="anchor"></a>Analyze data</h2>
<p>For this example, we created a small table with the relevant phenotype data for 12 samples: 6 from fetal samples and 6 from adult samples. We chose at random a brain region, in this case the primary auditory cortex (core) and for the example we will only look at data from chromosome 21. Other variables include the age in years and the gender. The data is shown below.</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"knitr"</span>)
<span class="co">## Get pheno table</span>
<span class="no">pheno</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span>(<span class="no">brainspanPheno</span>, <span class="no">structure_acronym</span> <span class="kw">==</span> <span class="st">"A1C"</span>)

<span class="co">## Display the main information</span>
<span class="no">p</span> <span class="kw">&lt;-</span> <span class="no">pheno</span>[, -<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span>(<span class="no">pheno</span>) <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(
    <span class="st">"structure_acronym"</span>,
    <span class="st">"structure_name"</span>, <span class="st">"file"</span>
))]
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">p</span>) <span class="kw">&lt;-</span> <span class="kw">NULL</span>
<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="no">p</span>, <span class="kw">format</span> <span class="kw">=</span> <span class="st">"html"</span>, <span class="kw">row.names</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<table class="table">
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
gender
</th>
<th style="text-align:left;">
lab
</th>
<th style="text-align:right;">
Age
</th>
<th style="text-align:left;">
group
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
HSB114.A1C
</td>
<td style="text-align:right;">
-0.5192308
</td>
<td style="text-align:left;">
fetal
</td>
</tr>
<tr>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
HSB103.A1C
</td>
<td style="text-align:right;">
-0.5192308
</td>
<td style="text-align:left;">
fetal
</td>
</tr>
<tr>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
HSB178.A1C
</td>
<td style="text-align:right;">
-0.4615385
</td>
<td style="text-align:left;">
fetal
</td>
</tr>
<tr>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
HSB154.A1C
</td>
<td style="text-align:right;">
-0.4615385
</td>
<td style="text-align:left;">
fetal
</td>
</tr>
<tr>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
F
</td>
<td style="text-align:left;">
HSB150.A1C
</td>
<td style="text-align:right;">
-0.5384615
</td>
<td style="text-align:left;">
fetal
</td>
</tr>
<tr>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
F
</td>
<td style="text-align:left;">
HSB149.A1C
</td>
<td style="text-align:right;">
-0.5192308
</td>
<td style="text-align:left;">
fetal
</td>
</tr>
<tr>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
F
</td>
<td style="text-align:left;">
HSB130.A1C
</td>
<td style="text-align:right;">
21.0000000
</td>
<td style="text-align:left;">
adult
</td>
</tr>
<tr>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
HSB136.A1C
</td>
<td style="text-align:right;">
23.0000000
</td>
<td style="text-align:left;">
adult
</td>
</tr>
<tr>
<td style="text-align:left;">
9
</td>
<td style="text-align:left;">
F
</td>
<td style="text-align:left;">
HSB126.A1C
</td>
<td style="text-align:right;">
30.0000000
</td>
<td style="text-align:left;">
adult
</td>
</tr>
<tr>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
HSB145.A1C
</td>
<td style="text-align:right;">
36.0000000
</td>
<td style="text-align:left;">
adult
</td>
</tr>
<tr>
<td style="text-align:left;">
11
</td>
<td style="text-align:left;">
M
</td>
<td style="text-align:left;">
HSB123.A1C
</td>
<td style="text-align:right;">
37.0000000
</td>
<td style="text-align:left;">
adult
</td>
</tr>
<tr>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
F
</td>
<td style="text-align:left;">
HSB135.A1C
</td>
<td style="text-align:right;">
40.0000000
</td>
<td style="text-align:left;">
adult
</td>
</tr>
</tbody>
</table>
<p>We can load the data from <em><a href="https://bioconductor.org/packages/3.11/derfinderData">derfinderData</a></em> <a id="cite-Collado-Torres_2020"></a>(<a href="https://github.com/leekgroup/derfinderData">Collado-Torres, Jaffe, and Leek, 2020</a>) by first identifying the paths to the BigWig files with <code><a href="https://rdrr.io/pkg/derfinder/man/rawFiles.html">derfinder::rawFiles()</a></code> and then loading the data with <code><a href="https://rdrr.io/pkg/derfinder/man/fullCoverage.html">derfinder::fullCoverage()</a></code>.</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="co">## Determine the files to use and fix the names</span>
<span class="no">files</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/rawFiles.html">rawFiles</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"A1C"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"derfinderData"</span>),
    <span class="kw">samplepatt</span> <span class="kw">=</span> <span class="st">"bw"</span>, <span class="kw">fileterm</span> <span class="kw">=</span> <span class="kw">NULL</span>
)
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">files</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span>(<span class="st">".bw"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">files</span>))

<span class="co">## Load the data from disk</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(<span class="no">fullCov</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/fullCoverage.html">fullCoverage</a></span>(<span class="kw">files</span> <span class="kw">=</span> <span class="no">files</span>, <span class="kw">chrs</span> <span class="kw">=</span> <span class="st">"chr21"</span>))</pre></body></html></div>
<pre><code>## 2020-04-27 03:23:57 fullCoverage: processing chromosome chr21</code></pre>
<pre><code>## 2020-04-27 03:23:57 loadCoverage: finding chromosome lengths</code></pre>
<pre><code>## 2020-04-27 03:23:57 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/A1C/HSB103.bw</code></pre>
<pre><code>## 2020-04-27 03:23:57 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/A1C/HSB114.bw</code></pre>
<pre><code>## 2020-04-27 03:23:57 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/A1C/HSB123.bw</code></pre>
<pre><code>## 2020-04-27 03:23:57 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/A1C/HSB126.bw</code></pre>
<pre><code>## 2020-04-27 03:23:57 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/A1C/HSB130.bw</code></pre>
<pre><code>## 2020-04-27 03:23:58 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/A1C/HSB135.bw</code></pre>
<pre><code>## 2020-04-27 03:23:58 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/A1C/HSB136.bw</code></pre>
<pre><code>## 2020-04-27 03:23:58 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/A1C/HSB145.bw</code></pre>
<pre><code>## 2020-04-27 03:23:58 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/A1C/HSB149.bw</code></pre>
<pre><code>## 2020-04-27 03:23:58 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/A1C/HSB150.bw</code></pre>
<pre><code>## 2020-04-27 03:23:58 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/A1C/HSB154.bw</code></pre>
<pre><code>## 2020-04-27 03:23:59 loadCoverage: loading BigWig file /usr/local/lib/R/host-site-library/derfinderData/extdata/A1C/HSB178.bw</code></pre>
<pre><code>## 2020-04-27 03:23:59 loadCoverage: applying the cutoff to the merged data</code></pre>
<pre><code>## 2020-04-27 03:23:59 filterData: originally there were 48129895 rows, now there are 48129895 rows. Meaning that 0 percent was filtered.</code></pre>
<pre><code>##    user  system elapsed 
##   2.711   0.049   2.762</code></pre>
<p>Alternatively, since the BigWig files are publicly available from <em>BrainSpan</em> (see <a href="http://download.alleninstitute.org/brainspan/MRF_BigWig_Gencode_v10/bigwig/">here</a>), we can extract the relevant coverage data using <code><a href="https://rdrr.io/pkg/derfinder/man/fullCoverage.html">derfinder::fullCoverage()</a></code>. Note that as of <em><a href="https://bioconductor.org/packages/3.11/rtracklayer">rtracklayer</a></em> 1.25.16 BigWig files are not supported on Windows: you can find the <code>fullCov</code> object inside <em><a href="https://bioconductor.org/packages/3.11/derfinderData">derfinderData</a></em> to follow the examples.</p>
<div class="sourceCode" id="cb25"><html><body><pre class="r"><span class="co">## Determine the files to use and fix the names</span>
<span class="no">files</span> <span class="kw">&lt;-</span> <span class="no">pheno</span>$<span class="no">file</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">files</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span>(<span class="st">".A1C"</span>, <span class="st">""</span>, <span class="no">pheno</span>$<span class="no">lab</span>)

<span class="co">## Load the data from the web</span>
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(<span class="no">fullCov</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/fullCoverage.html">fullCoverage</a></span>(<span class="kw">files</span> <span class="kw">=</span> <span class="no">files</span>, <span class="kw">chrs</span> <span class="kw">=</span> <span class="st">"chr21"</span>))</pre></body></html></div>
<p>Once we have the base-level coverage data for all 12 samples, we can construct the models. In this case, we want to find differences between fetal and adult samples while adjusting for gender and a proxy of the library size.</p>
<div class="sourceCode" id="cb26"><html><body><pre class="r"><span class="co">## Get some idea of the library sizes</span>
<span class="no">sampleDepths</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/sampleDepth.html">sampleDepth</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/collapseFullCoverage.html">collapseFullCoverage</a></span>(<span class="no">fullCov</span>), <span class="fl">1</span>)</pre></body></html></div>
<pre><code>## 2020-04-27 03:23:59 sampleDepth: Calculating sample quantiles</code></pre>
<pre><code>## 2020-04-27 03:24:00 sampleDepth: Calculating sample adjustments</code></pre>
<div class="sourceCode" id="cb29"><html><body><pre class="r"><span class="co">## Define models</span>
<span class="no">models</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/makeModels.html">makeModels</a></span>(<span class="no">sampleDepths</span>,
    <span class="kw">testvars</span> <span class="kw">=</span> <span class="no">pheno</span>$<span class="no">group</span>,
    <span class="kw">adjustvars</span> <span class="kw">=</span> <span class="no">pheno</span>[, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"gender"</span>)]
)</pre></body></html></div>
<p>Next, we can find candidate differentially expressed regions (DERs) using as input the segments of the genome where at least one sample has coverage greater than 3. In this particular example, we chose a low theoretical F-statistic cutoff and used 20 permutations.</p>
<div class="sourceCode" id="cb30"><html><body><pre class="r"><span class="co">## Filter coverage</span>
<span class="no">filteredCov</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="no">fullCov</span>, <span class="no">filterData</span>, <span class="kw">cutoff</span> <span class="kw">=</span> <span class="fl">3</span>)</pre></body></html></div>
<pre><code>## 2020-04-27 03:24:01 filterData: originally there were 48129895 rows, now there are 90023 rows. Meaning that 99.81 percent was filtered.</code></pre>
<div class="sourceCode" id="cb32"><html><body><pre class="r"><span class="co">## Perform differential expression analysis</span>
<span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"bumphunter"</span>))
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(<span class="no">results</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/analyzeChr.html">analyzeChr</a></span>(
    <span class="kw">chr</span> <span class="kw">=</span> <span class="st">"chr21"</span>, <span class="no">filteredCov</span>$<span class="no">chr21</span>,
    <span class="no">models</span>, <span class="kw">groupInfo</span> <span class="kw">=</span> <span class="no">pheno</span>$<span class="no">group</span>, <span class="kw">writeOutput</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
    <span class="kw">cutoffFstat</span> <span class="kw">=</span> <span class="fl">5e-02</span>, <span class="kw">nPermute</span> <span class="kw">=</span> <span class="fl">20</span>, <span class="kw">seeds</span> <span class="kw">=</span> <span class="fl">20140923</span> + <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span>(<span class="fl">20</span>)
))</pre></body></html></div>
<pre><code>## 2020-04-27 03:24:02 analyzeChr: Pre-processing the coverage data</code></pre>
<pre><code>## 2020-04-27 03:24:04 analyzeChr: Calculating statistics</code></pre>
<pre><code>## 2020-04-27 03:24:04 calculateStats: calculating the F-statistics</code></pre>
<pre><code>## 2020-04-27 03:24:04 analyzeChr: Calculating pvalues</code></pre>
<pre><code>## 2020-04-27 03:24:04 analyzeChr: Using the following theoretical cutoff for the F-statistics 5.31765507157871</code></pre>
<pre><code>## 2020-04-27 03:24:04 calculatePvalues: identifying data segments</code></pre>
<pre><code>## 2020-04-27 03:24:04 findRegions: segmenting information</code></pre>
<pre><code>## 2020-04-27 03:24:04 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-04-27 03:24:04 findRegions: identifying region clusters</code></pre>
<pre><code>## 2020-04-27 03:24:04 calculatePvalues: calculating F-statistics for permutation 1 and seed 20140924</code></pre>
<pre><code>## 2020-04-27 03:24:04 findRegions: segmenting information</code></pre>
<pre><code>## 2020-04-27 03:24:04 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-04-27 03:24:04 calculatePvalues: calculating F-statistics for permutation 2 and seed 20140925</code></pre>
<pre><code>## 2020-04-27 03:24:04 findRegions: segmenting information</code></pre>
<pre><code>## 2020-04-27 03:24:04 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-04-27 03:24:04 calculatePvalues: calculating F-statistics for permutation 3 and seed 20140926</code></pre>
<pre><code>## 2020-04-27 03:24:05 findRegions: segmenting information</code></pre>
<pre><code>## 2020-04-27 03:24:05 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-04-27 03:24:05 calculatePvalues: calculating F-statistics for permutation 4 and seed 20140927</code></pre>
<pre><code>## 2020-04-27 03:24:05 findRegions: segmenting information</code></pre>
<pre><code>## 2020-04-27 03:24:05 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-04-27 03:24:05 calculatePvalues: calculating F-statistics for permutation 5 and seed 20140928</code></pre>
<pre><code>## 2020-04-27 03:24:05 findRegions: segmenting information</code></pre>
<pre><code>## 2020-04-27 03:24:05 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-04-27 03:24:05 calculatePvalues: calculating F-statistics for permutation 6 and seed 20140929</code></pre>
<pre><code>## 2020-04-27 03:24:05 findRegions: segmenting information</code></pre>
<pre><code>## 2020-04-27 03:24:05 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-04-27 03:24:05 calculatePvalues: calculating F-statistics for permutation 7 and seed 20140930</code></pre>
<pre><code>## 2020-04-27 03:24:05 findRegions: segmenting information</code></pre>
<pre><code>## 2020-04-27 03:24:05 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-04-27 03:24:05 calculatePvalues: calculating F-statistics for permutation 8 and seed 20140931</code></pre>
<pre><code>## 2020-04-27 03:24:06 findRegions: segmenting information</code></pre>
<pre><code>## 2020-04-27 03:24:06 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-04-27 03:24:06 calculatePvalues: calculating F-statistics for permutation 9 and seed 20140932</code></pre>
<pre><code>## 2020-04-27 03:24:06 findRegions: segmenting information</code></pre>
<pre><code>## 2020-04-27 03:24:06 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-04-27 03:24:06 calculatePvalues: calculating F-statistics for permutation 10 and seed 20140933</code></pre>
<pre><code>## 2020-04-27 03:24:06 findRegions: segmenting information</code></pre>
<pre><code>## 2020-04-27 03:24:06 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-04-27 03:24:06 calculatePvalues: calculating F-statistics for permutation 11 and seed 20140934</code></pre>
<pre><code>## 2020-04-27 03:24:06 findRegions: segmenting information</code></pre>
<pre><code>## 2020-04-27 03:24:06 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-04-27 03:24:06 calculatePvalues: calculating F-statistics for permutation 12 and seed 20140935</code></pre>
<pre><code>## 2020-04-27 03:24:06 findRegions: segmenting information</code></pre>
<pre><code>## 2020-04-27 03:24:06 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-04-27 03:24:06 calculatePvalues: calculating F-statistics for permutation 13 and seed 20140936</code></pre>
<pre><code>## 2020-04-27 03:24:06 findRegions: segmenting information</code></pre>
<pre><code>## 2020-04-27 03:24:06 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-04-27 03:24:07 calculatePvalues: calculating F-statistics for permutation 14 and seed 20140937</code></pre>
<pre><code>## 2020-04-27 03:24:07 findRegions: segmenting information</code></pre>
<pre><code>## 2020-04-27 03:24:07 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-04-27 03:24:07 calculatePvalues: calculating F-statistics for permutation 15 and seed 20140938</code></pre>
<pre><code>## 2020-04-27 03:24:07 findRegions: segmenting information</code></pre>
<pre><code>## 2020-04-27 03:24:07 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-04-27 03:24:07 calculatePvalues: calculating F-statistics for permutation 16 and seed 20140939</code></pre>
<pre><code>## 2020-04-27 03:24:07 findRegions: segmenting information</code></pre>
<pre><code>## 2020-04-27 03:24:07 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-04-27 03:24:07 calculatePvalues: calculating F-statistics for permutation 17 and seed 20140940</code></pre>
<pre><code>## 2020-04-27 03:24:07 findRegions: segmenting information</code></pre>
<pre><code>## 2020-04-27 03:24:07 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-04-27 03:24:07 calculatePvalues: calculating F-statistics for permutation 18 and seed 20140941</code></pre>
<pre><code>## 2020-04-27 03:24:07 findRegions: segmenting information</code></pre>
<pre><code>## 2020-04-27 03:24:07 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-04-27 03:24:07 calculatePvalues: calculating F-statistics for permutation 19 and seed 20140942</code></pre>
<pre><code>## 2020-04-27 03:24:08 findRegions: segmenting information</code></pre>
<pre><code>## 2020-04-27 03:24:08 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-04-27 03:24:08 calculatePvalues: calculating F-statistics for permutation 20 and seed 20140943</code></pre>
<pre><code>## 2020-04-27 03:24:08 findRegions: segmenting information</code></pre>
<pre><code>## 2020-04-27 03:24:08 findRegions: identifying candidate regions</code></pre>
<pre><code>## 2020-04-27 03:24:08 calculatePvalues: calculating the p-values</code></pre>
<pre><code>## 2020-04-27 03:24:08 analyzeChr: Annotating regions</code></pre>
<pre><code>## No annotationPackage supplied. Trying org.Hs.eg.db.</code></pre>
<pre><code>## Loading required package: org.Hs.eg.db</code></pre>
<pre><code>## Loading required package: AnnotationDbi</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
<pre><code>## </code></pre>
<pre><code>## Getting TSS and TSE.</code></pre>
<pre><code>## Getting CSS and CSE.</code></pre>
<pre><code>## Getting exons.</code></pre>
<pre><code>## Annotating genes.</code></pre>
<pre><code>## ...</code></pre>
<pre><code>##    user  system elapsed 
##  44.777   2.909  44.230</code></pre>
<div class="sourceCode" id="cb116"><html><body><pre class="r"><span class="co">## Quick access to the results</span>
<span class="no">regions</span> <span class="kw">&lt;-</span> <span class="no">results</span>$<span class="no">regions</span>$<span class="no">regions</span>

<span class="co">## Annotation database to use</span>
<span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"TxDb.Hsapiens.UCSC.hg19.knownGene"</span>))
<span class="no">txdb</span> <span class="kw">&lt;-</span> <span class="no">TxDb.Hsapiens.UCSC.hg19.knownGene</span></pre></body></html></div>
</div>
<div id="plotoverview" class="section level2">
<h2 class="hasAnchor">
<a href="#plotoverview" class="anchor"></a><code>plotOverview()</code>
</h2>
<p>Now that we have obtained the main results using <em><a href="https://bioconductor.org/packages/3.11/derfinder">derfinder</a></em>, we can proceed to visualizing the results using <em><a href="https://bioconductor.org/packages/3.11/derfinderPlot">derfinderPlot</a></em>. The easiest to use of all the functions is <code><a href="../reference/plotOverview.html">plotOverview()</a></code> which takes a set of regions and annotation information produced by <code><a href="https://rdrr.io/pkg/bumphunter/man/matchGenes.html">bumphunter::matchGenes()</a></code>.</p>
<p>Figure @ref(fig:plotOverview) shows the candidate DERs colored by whether their q-value was less than 0.10 or not.</p>
<div class="sourceCode" id="cb117"><html><body><pre class="r"><span class="co">## Q-values overview</span>
<span class="fu"><a href="../reference/plotOverview.html">plotOverview</a></span>(<span class="kw">regions</span> <span class="kw">=</span> <span class="no">regions</span>, <span class="kw">annotation</span> <span class="kw">=</span> <span class="no">results</span>$<span class="no">annotation</span>, <span class="kw">type</span> <span class="kw">=</span> <span class="st">"qval"</span>)</pre></body></html></div>
<pre><code>## 2020-04-27 03:24:46 plotOverview: assigning chromosome lengths from hg19!</code></pre>
<pre><code>## Scale for 'x' is already present. Adding another scale for 'x', which will
## replace the existing scale.
## Scale for 'x' is already present. Adding another scale for 'x', which will
## replace the existing scale.</code></pre>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotOverview-1.png" alt="Location of the DERs in the genome. This plot is was designed for many chromosomes but only one is shown here for simplicity." width="700"><p class="caption">
Location of the DERs in the genome. This plot is was designed for many chromosomes but only one is shown here for simplicity.
</p>
</div>
<p>Figure @ref(fig:plotOverview2) shows the candidate DERs colored by the type of gene feature they are nearest too.</p>
<div class="sourceCode" id="cb120"><html><body><pre class="r"><span class="co">## Annotation overview</span>
<span class="fu"><a href="../reference/plotOverview.html">plotOverview</a></span>(
    <span class="kw">regions</span> <span class="kw">=</span> <span class="no">regions</span>, <span class="kw">annotation</span> <span class="kw">=</span> <span class="no">results</span>$<span class="no">annotation</span>,
    <span class="kw">type</span> <span class="kw">=</span> <span class="st">"annotation"</span>
)</pre></body></html></div>
<pre><code>## 2020-04-27 03:24:47 plotOverview: assigning chromosome lengths from hg19!</code></pre>
<pre><code>## Scale for 'x' is already present. Adding another scale for 'x', which will
## replace the existing scale.</code></pre>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotOverview2-1.png" alt="Location of the DERs in the genome and colored by annotation class. This plot is was designed for many chromosomes but only one is shown here for simplicity." width="700"><p class="caption">
Location of the DERs in the genome and colored by annotation class. This plot is was designed for many chromosomes but only one is shown here for simplicity.
</p>
</div>
<p>In this particular example, because we are only using data from one chromosome the above plot is not as informative as in a real case scenario. However, with this plot we can quickly observe that nearly all of the candidate DERs are inside an exon.</p>
</div>
<div id="plotregioncoverage" class="section level2">
<h2 class="hasAnchor">
<a href="#plotregioncoverage" class="anchor"></a><code>plotRegionCoverage()</code>
</h2>
<p>The complete opposite of visualizing the candidate DERs at the genome-level is to visualize them one region at a time. <code><a href="../reference/plotRegionCoverage.html">plotRegionCoverage()</a></code> allows us to do this quickly for a large number of regions.</p>
<p>Before using this function, we need to process more detailed information using two <em><a href="https://bioconductor.org/packages/3.11/derfinder">derfinder</a></em> functions: <code><a href="https://rdrr.io/pkg/derfinder/man/annotateRegions.html">annotateRegions()</a></code> and <code><a href="https://rdrr.io/pkg/derfinder/man/getRegionCoverage.html">getRegionCoverage()</a></code> as shown below.</p>
<div class="sourceCode" id="cb123"><html><body><pre class="r"><span class="co">## Get required information for the plots</span>
<span class="no">annoRegs</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/annotateRegions.html">annotateRegions</a></span>(<span class="no">regions</span>, <span class="no">genomicState</span>$<span class="no">fullGenome</span>)</pre></body></html></div>
<pre><code>## 2020-04-27 03:24:48 annotateRegions: counting</code></pre>
<pre><code>## 2020-04-27 03:24:48 annotateRegions: annotating</code></pre>
<div class="sourceCode" id="cb126"><html><body><pre class="r"><span class="no">regionCov</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/derfinder/man/getRegionCoverage.html">getRegionCoverage</a></span>(<span class="no">fullCov</span>, <span class="no">regions</span>)</pre></body></html></div>
<pre><code>## 2020-04-27 03:24:48 getRegionCoverage: processing chr21</code></pre>
<pre><code>## 2020-04-27 03:24:48 getRegionCoverage: done processing chr21</code></pre>
<p>Once we have the relevant information we can proceed to plotting the first 10 regions. In this case, we will supply <code><a href="../reference/plotRegionCoverage.html">plotRegionCoverage()</a></code> with the information it needs to plot transcripts overlapping these 10 regions (Figures @ref(fig:plotRegCov1), @ref(fig:plotRegCov2), @ref(fig:plotRegCov3), @ref(fig:plotRegCov4), @ref(fig:plotRegCov5), @ref(fig:plotRegCov6), @ref(fig:plotRegCov7), @ref(fig:plotRegCov8), @ref(fig:plotRegCov9), @ref(fig:plotRegCov10)).</p>
<div class="sourceCode" id="cb129"><html><body><pre class="r"><span class="co">## Plot top 10 regions</span>
<span class="fu"><a href="../reference/plotRegionCoverage.html">plotRegionCoverage</a></span>(
    <span class="kw">regions</span> <span class="kw">=</span> <span class="no">regions</span>, <span class="kw">regionCoverage</span> <span class="kw">=</span> <span class="no">regionCov</span>,
    <span class="kw">groupInfo</span> <span class="kw">=</span> <span class="no">pheno</span>$<span class="no">group</span>, <span class="kw">nearestAnnotation</span> <span class="kw">=</span> <span class="no">results</span>$<span class="no">annotation</span>,
    <span class="kw">annotatedRegions</span> <span class="kw">=</span> <span class="no">annoRegs</span>, <span class="kw">whichRegions</span> <span class="kw">=</span> <span class="fl">1</span>:<span class="fl">10</span>, <span class="kw">txdb</span> <span class="kw">=</span> <span class="no">txdb</span>, <span class="kw">scalefac</span> <span class="kw">=</span> <span class="fl">1</span>,
    <span class="kw">ask</span> <span class="kw">=</span> <span class="fl">FALSE</span>, <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>
)</pre></body></html></div>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotRegCov-1.png" alt="Base-pair resolution plot of differentially expressed region 1." width="700"><p class="caption">
Base-pair resolution plot of differentially expressed region 1.
</p>
</div>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotRegCov-2.png" alt="Base-pair resolution plot of differentially expressed region 2." width="700"><p class="caption">
Base-pair resolution plot of differentially expressed region 2.
</p>
</div>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotRegCov-3.png" alt="Base-pair resolution plot of differentially expressed region 3." width="700"><p class="caption">
Base-pair resolution plot of differentially expressed region 3.
</p>
</div>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotRegCov-4.png" alt="Base-pair resolution plot of differentially expressed region 4." width="700"><p class="caption">
Base-pair resolution plot of differentially expressed region 4.
</p>
</div>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotRegCov-5.png" alt="Base-pair resolution plot of differentially expressed region 5." width="700"><p class="caption">
Base-pair resolution plot of differentially expressed region 5.
</p>
</div>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotRegCov-6.png" alt="Base-pair resolution plot of differentially expressed region 6." width="700"><p class="caption">
Base-pair resolution plot of differentially expressed region 6.
</p>
</div>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotRegCov-7.png" alt="Base-pair resolution plot of differentially expressed region 7." width="700"><p class="caption">
Base-pair resolution plot of differentially expressed region 7.
</p>
</div>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotRegCov-8.png" alt="Base-pair resolution plot of differentially expressed region 8." width="700"><p class="caption">
Base-pair resolution plot of differentially expressed region 8.
</p>
</div>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotRegCov-9.png" alt="Base-pair resolution plot of differentially expressed region 9." width="700"><p class="caption">
Base-pair resolution plot of differentially expressed region 9.
</p>
</div>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotRegCov-10.png" alt="Base-pair resolution plot of differentially expressed region 10." width="700"><p class="caption">
Base-pair resolution plot of differentially expressed region 10.
</p>
</div>
<p>The base-level coverage is shown in a log2 scale with any overlapping exons shown in dark blue and known introns in light blue.</p>
</div>
<div id="plotcluster" class="section level2">
<h2 class="hasAnchor">
<a href="#plotcluster" class="anchor"></a><code>plotCluster()</code>
</h2>
<p>In this example, we noticed with the <code><a href="../reference/plotRegionCoverage.html">plotRegionCoverage()</a></code> plots that most of the candidate DERs are contained in known exons. Sometimes, the signal might be low or we might have used very stringent cutoffs in the <em><a href="https://bioconductor.org/packages/3.11/derfinder">derfinder</a></em> analysis. One way we can observe this is by plotting clusters of regions where a cluster is defined as regions within 300 bp (default option) of each other.</p>
<p>To visualize the clusters, we can use <code><a href="../reference/plotCluster.html">plotCluster()</a></code> which takes similar input to <code><a href="../reference/plotOverview.html">plotOverview()</a></code> with the notable addition of the coverage information as well as the <code>idx</code> argument. This argument specifies which region to focus on: it will be plotted with a red bar and will determine the cluster to display.</p>
<p>In Figure @ref(fig:plotCluster) we observe one large candidate DER with other nearby ones that do not have a q-value less than 0.10. In a real analysis, we would probably discard this region as the coverage is fairly low.</p>
<div class="sourceCode" id="cb130"><html><body><pre class="r"><span class="co">## First cluster</span>
<span class="fu"><a href="../reference/plotCluster.html">plotCluster</a></span>(
    <span class="kw">idx</span> <span class="kw">=</span> <span class="fl">1</span>, <span class="kw">regions</span> <span class="kw">=</span> <span class="no">regions</span>, <span class="kw">annotation</span> <span class="kw">=</span> <span class="no">results</span>$<span class="no">annotation</span>,
    <span class="kw">coverageInfo</span> <span class="kw">=</span> <span class="no">fullCov</span>$<span class="no">chr21</span>, <span class="kw">txdb</span> <span class="kw">=</span> <span class="no">txdb</span>, <span class="kw">groupInfo</span> <span class="kw">=</span> <span class="no">pheno</span>$<span class="no">group</span>,
    <span class="kw">titleUse</span> <span class="kw">=</span> <span class="st">"pval"</span>
)</pre></body></html></div>
<pre><code>## Parsing transcripts...</code></pre>
<pre><code>## Parsing exons...</code></pre>
<pre><code>## Parsing cds...</code></pre>
<pre><code>## Parsing utrs...</code></pre>
<pre><code>## ------exons...</code></pre>
<pre><code>## ------cdss...</code></pre>
<pre><code>## ------introns...</code></pre>
<pre><code>## ------utr...</code></pre>
<pre><code>## aggregating...</code></pre>
<pre><code>## Done</code></pre>
<pre><code>## Constructing graphics...</code></pre>
<pre><code>## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.</code></pre>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotCluster-1.png" alt="Cluster plot for cluster 1 using ggbio." width="700"><p class="caption">
Cluster plot for cluster 1 using ggbio.
</p>
</div>
<p>The second cluster (Figure @ref(fig:plotCluster2)) shows a larger number of potential DERs (again without q-values less than 0.10) in a segment of the genome where the coverage data is highly variable. This is a common occurrence with RNA-seq data.</p>
<div class="sourceCode" id="cb143"><html><body><pre class="r"><span class="co">## Second cluster</span>
<span class="fu"><a href="../reference/plotCluster.html">plotCluster</a></span>(
    <span class="kw">idx</span> <span class="kw">=</span> <span class="fl">2</span>, <span class="kw">regions</span> <span class="kw">=</span> <span class="no">regions</span>, <span class="kw">annotation</span> <span class="kw">=</span> <span class="no">results</span>$<span class="no">annotation</span>,
    <span class="kw">coverageInfo</span> <span class="kw">=</span> <span class="no">fullCov</span>$<span class="no">chr21</span>, <span class="kw">txdb</span> <span class="kw">=</span> <span class="no">txdb</span>, <span class="kw">groupInfo</span> <span class="kw">=</span> <span class="no">pheno</span>$<span class="no">group</span>,
    <span class="kw">titleUse</span> <span class="kw">=</span> <span class="st">"pval"</span>
)</pre></body></html></div>
<pre><code>## Warning in getIdeoGR(data): geom(ideogram) need valid seqlengths information for accurate mapping,
##                  now use reduced information as ideogram...</code></pre>
<pre><code>## Warning: `panel.margin` is deprecated. Please use `panel.spacing` property
## instead

## Warning: `panel.margin` is deprecated. Please use `panel.spacing` property
## instead</code></pre>
<pre><code>## Parsing transcripts...</code></pre>
<pre><code>## Parsing exons...</code></pre>
<pre><code>## Parsing cds...</code></pre>
<pre><code>## Parsing utrs...</code></pre>
<pre><code>## ------exons...</code></pre>
<pre><code>## ------cdss...</code></pre>
<pre><code>## ------introns...</code></pre>
<pre><code>## ------utr...</code></pre>
<pre><code>## aggregating...</code></pre>
<pre><code>## Done</code></pre>
<pre><code>## Constructing graphics...</code></pre>
<pre><code>## Warning: `panel.margin` is deprecated. Please use `panel.spacing` property
## instead

## Warning: `panel.margin` is deprecated. Please use `panel.spacing` property
## instead

## Warning: `panel.margin` is deprecated. Please use `panel.spacing` property
## instead

## Warning: `panel.margin` is deprecated. Please use `panel.spacing` property
## instead

## Warning: `panel.margin` is deprecated. Please use `panel.spacing` property
## instead

## Warning: `panel.margin` is deprecated. Please use `panel.spacing` property
## instead

## Warning: `panel.margin` is deprecated. Please use `panel.spacing` property
## instead

## Warning: `panel.margin` is deprecated. Please use `panel.spacing` property
## instead

## Warning: `panel.margin` is deprecated. Please use `panel.spacing` property
## instead</code></pre>
<pre><code>## Coordinate system already present. Adding new coordinate system, which will replace the existing one.</code></pre>
<pre><code>## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.
## Coordinate system already present. Adding new coordinate system, which will replace the existing one.</code></pre>
<pre><code>## Warning: Transformation introduced infinite values in continuous y-axis</code></pre>
<div class="figure">
<img src="derfinderPlot_files/figure-html/plotCluster2-1.png" alt="Cluster plot for cluster 2 using ggbio." width="700"><p class="caption">
Cluster plot for cluster 2 using ggbio.
</p>
</div>
<p>These plots show an ideogram which helps quickly identify which region of the genome we are focusing on. Then, the base-level coverage information for each sample is displayed in log2. Next, the coverage group means are shown in the log2 scale. The plot is completed with the potential and candidate DERs as well as any known transcripts.</p>
</div>
<div id="vennregions" class="section level2">
<h2 class="hasAnchor">
<a href="#vennregions" class="anchor"></a><code>vennRegions</code>
</h2>
<p><em><a href="https://bioconductor.org/packages/3.11/derfinder">derfinder</a></em> has functions for annotating regions given their genomic state. A typical visualization is to then view how many regions overlap known exons, introns, intergenic regions, none of them or several of these groups in a venn diagram. The function <code><a href="../reference/vennRegions.html">vennRegions()</a></code> makes this plot using the output from <code><a href="https://rdrr.io/pkg/derfinder/man/annotateRegions.html">derfinder::annotateRegions()</a></code> as shown in Figure @ref(fig:vennRegions).</p>
<div class="sourceCode" id="cb161"><html><body><pre class="r"><span class="co">## Make venn diagram</span>
<span class="no">venn</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/vennRegions.html">vennRegions</a></span>(<span class="no">annoRegs</span>)</pre></body></html></div>
<div class="figure">
<img src="derfinderPlot_files/figure-html/vennRegions-1.png" alt="Venn diagram of regions by annotation class." width="700"><p class="caption">
Venn diagram of regions by annotation class.
</p>
</div>
<div class="sourceCode" id="cb162"><html><body><pre class="r"><span class="co">## It returns the actual venn counts information</span>
<span class="no">venn</span></pre></body></html></div>
<pre><code>##   exon intergenic intron Counts
## 1    0          0      0      0
## 2    0          0      1      2
## 3    0          1      0      4
## 4    0          1      1      0
## 5    1          0      0    259
## 6    1          0      1     35
## 7    1          1      0      0
## 8    1          1      1      0
## attr(,"class")
## [1] "VennCounts"</code></pre>
</div>
</div>
<div id="reproducibility" class="section level1">
<h1 class="hasAnchor">
<a href="#reproducibility" class="anchor"></a>Reproducibility</h1>
<p>This package was made possible thanks to:</p>
<ul>
<li>R <a id="cite-R_Core_Team_2020"></a>(<a href="https://www.R-project.org/">R Core Team, 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.11/GenomeInfoDb">GenomeInfoDb</a></em> <a id="cite-GenomeInfoDb"></a>(<a href="https://doi.org/10.18129/B9.bioc.GenomeInfoDb">Arora, Morgan, Carlson, and Pagès, 2017</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.11/GenomicRanges">GenomicRanges</a></em> <a id="cite-Lawrence_2013"></a>(<a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118">Lawrence, Huber, Pagès, Aboyoun, et al., 2013</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.11/ggbio">ggbio</a></em> (<a href="#bib-Yin_2012">Yin, Cook, and Lawrence, 2012</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a></em> <a id="cite-Wickham_2016"></a>(<a href="https://ggplot2.tidyverse.org">Wickham, 2016</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.11/IRanges">IRanges</a></em> <a id="cite-Lawrence_2013a"></a>(<a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118">Lawrence, Huber, Pagès, Aboyoun, et al., 2013</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=plyr">plyr</a></em> <a id="cite-Wickham_2011"></a>(<a href="http://www.jstatsoft.org/v40/i01/">Wickham, 2011</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RColorBrewer">RColorBrewer</a></em> <a id="cite-Neuwirth_2014"></a>(<a href="https://CRAN.R-project.org/package=RColorBrewer">Neuwirth, 2014</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=reshape2">reshape2</a></em> <a id="cite-Wickham_2007"></a>(<a href="http://www.jstatsoft.org/v21/i12/">Wickham, 2007</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=scales">scales</a></em> <a id="cite-Wickham_2019"></a>(<a href="https://CRAN.R-project.org/package=scales">Wickham and Seidel, 2019</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.11/biovizBase">biovizBase</a></em> <a id="cite-Yin_2019"></a>(<a href="#bib-Yin_2019">Yin, Lawrence, and Cook, 2019</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.11/bumphunter">bumphunter</a></em> <a id="cite-bumphunter"></a>(<a href="https://doi.org/10.1093/ije/dyr238">Jaffe, Murakami, Lee, Leek, et al., 2012</a>) and <a id="cite-bumphunterPaper"></a>(<a href="#bib-bumphunterPaper">Jaffe, Murakami, Lee, Leek, et al., 2012</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.11/derfinder">derfinder</a></em> (<a href="http://nar.oxfordjournals.org/content/early/2016/09/29/nar.gkw852">Collado-Torres, Nellore, Frazee, Wilks, et al., 2017</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.11/derfinderData">derfinderData</a></em> (<a href="https://github.com/leekgroup/derfinderData">Collado-Torres, Jaffe, and Leek, 2020</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo">sessioninfo</a></em> <a id="cite-Csardi_2018"></a>(<a href="https://CRAN.R-project.org/package=sessioninfo">Csárdi, core, Wickham, Chang, et al., 2018</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitcitations">knitcitations</a></em> <a id="cite-Boettiger_2019"></a>(<a href="https://CRAN.R-project.org/package=knitcitations">Boettiger, 2019</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr">knitr</a></em> <a id="cite-Xie_2014"></a>(<a href="http://www.crcpress.com/product/isbn/9781466561595">Xie, 2014</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.11/BiocStyle">BiocStyle</a></em> <a id="cite-Oles_2020"></a>(<a href="https://github.com/Bioconductor/BiocStyle">Oleś, Morgan, and Huber, 2020</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em> <a id="cite-Allaire_2020"></a>(<a href="https://github.com/rstudio/rmarkdown">Allaire, Xie, McPherson, Luraschi, et al., 2020</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat">testthat</a></em> <a id="cite-Wickham_2011a"></a>(<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf">Wickham, 2011</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.11/TxDb.Hsapiens.UCSC.hg19.knownGene">TxDb.Hsapiens.UCSC.hg19.knownGene</a></em> <a id="cite-Carlson_2015"></a>(<a href="#bib-Carlson_2015">Carlson and Maintainer, 2015</a>)</li>
</ul>
<p>Code for creating the vignette</p>
<div class="sourceCode" id="cb164"><html><body><pre class="r"><span class="co">## Create the vignette</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"rmarkdown"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/rmarkdown/man/render.html">render</a></span>(<span class="st">"derfinderPlot.Rmd"</span>, <span class="st">"BiocStyle::html_document"</span>))

<span class="co">## Extract the R code</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"knitr"</span>)
<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/knit.html">knit</a></span>(<span class="st">"derfinderPlot.Rmd"</span>, <span class="kw">tangle</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<div class="sourceCode" id="cb165"><html><body><pre class="r"><span class="co">## Clean up</span>
<span class="fu"><a href="https://rdrr.io/r/base/unlink.html">unlink</a></span>(<span class="st">"chr21"</span>, <span class="kw">recursive</span> <span class="kw">=</span> <span class="fl">TRUE</span>)
<span class="fu"><a href="https://rdrr.io/r/base/files.html">file.remove</a></span>(<span class="st">"derfinderPlotRef.bib"</span>)</pre></body></html></div>
<pre><code>## [1] TRUE</code></pre>
<p>Date the vignette was generated.</p>
<pre><code>## [1] "2020-04-27 03:25:12 UTC"</code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code>## Time difference of 1.499 mins</code></pre>
<p><code>R</code> session information.</p>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value                                             
##  version  R Under development (unstable) (2020-04-24 r78289)
##  os       Debian GNU/Linux 10 (buster)                      
##  system   x86_64, linux-gnu                                 
##  ui       X11                                               
##  language (EN)                                              
##  collate  en_US.UTF-8                                       
##  ctype    en_US.UTF-8                                       
##  tz       UTC                                               
##  date     2020-04-27                                        
## 
## ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package                           * version  date       lib source        
##  acepack                             1.4.1    2016-10-29 [1] CRAN (R 4.1.0)
##  AnnotationDbi                     * 1.49.1   2020-01-25 [1] Bioconductor  
##  AnnotationFilter                    1.11.0   2019-10-29 [1] Bioconductor  
##  askpass                             1.1      2019-01-13 [2] CRAN (R 4.1.0)
##  assertthat                          0.2.1    2019-03-21 [2] CRAN (R 4.1.0)
##  backports                           1.1.6    2020-04-05 [2] CRAN (R 4.1.0)
##  base64enc                           0.1-3    2015-07-28 [1] CRAN (R 4.1.0)
##  bibtex                              0.4.2.2  2020-01-02 [1] CRAN (R 4.1.0)
##  Biobase                           * 2.47.3   2020-03-16 [1] Bioconductor  
##  BiocFileCache                       1.11.6   2020-04-16 [1] Bioconductor  
##  BiocGenerics                      * 0.33.3   2020-03-23 [1] Bioconductor  
##  BiocManager                         1.30.10  2019-11-16 [2] CRAN (R 4.1.0)
##  BiocParallel                        1.21.3   2020-04-21 [1] Bioconductor  
##  BiocStyle                         * 2.15.8   2020-04-19 [1] Bioconductor  
##  biomaRt                             2.43.6   2020-04-21 [1] Bioconductor  
##  Biostrings                          2.55.7   2020-03-24 [1] Bioconductor  
##  biovizBase                          1.35.1   2019-12-03 [1] Bioconductor  
##  bit                                 1.1-15.2 2020-02-10 [1] CRAN (R 4.1.0)
##  bit64                               0.9-7    2017-05-08 [1] CRAN (R 4.1.0)
##  bitops                              1.0-6    2013-08-17 [1] CRAN (R 4.1.0)
##  blob                                1.2.1    2020-01-20 [1] CRAN (R 4.1.0)
##  bookdown                            0.18     2020-03-05 [1] CRAN (R 4.1.0)
##  BSgenome                            1.55.4   2020-03-19 [1] Bioconductor  
##  bumphunter                        * 1.29.0   2019-10-29 [1] Bioconductor  
##  checkmate                           2.0.0    2020-02-06 [1] CRAN (R 4.1.0)
##  cli                                 2.0.2    2020-02-28 [2] CRAN (R 4.1.0)
##  cluster                             2.1.0    2019-06-19 [3] CRAN (R 4.1.0)
##  codetools                           0.2-16   2018-12-24 [3] CRAN (R 4.1.0)
##  colorspace                          1.4-1    2019-03-18 [1] CRAN (R 4.1.0)
##  crayon                              1.3.4    2017-09-16 [2] CRAN (R 4.1.0)
##  curl                                4.3      2019-12-02 [2] CRAN (R 4.1.0)
##  data.table                          1.12.8   2019-12-09 [1] CRAN (R 4.1.0)
##  DBI                                 1.1.0    2019-12-15 [1] CRAN (R 4.1.0)
##  dbplyr                              1.4.3    2020-04-19 [1] CRAN (R 4.1.0)
##  DelayedArray                        0.13.12  2020-04-10 [1] Bioconductor  
##  derfinder                         * 1.21.7   2020-04-22 [1] Bioconductor  
##  derfinderData                     * 2.5.3    2020-04-23 [1] Bioconductor  
##  derfinderHelper                     1.21.2   2020-04-18 [1] Bioconductor  
##  derfinderPlot                     * 1.21.5   2020-04-27 [1] Bioconductor  
##  desc                                1.2.0    2018-05-01 [2] CRAN (R 4.1.0)
##  dichromat                           2.0-0    2013-01-24 [1] CRAN (R 4.1.0)
##  digest                              0.6.25   2020-02-23 [2] CRAN (R 4.1.0)
##  doRNG                               1.8.2    2020-01-27 [1] CRAN (R 4.1.0)
##  dplyr                               0.8.5    2020-03-07 [1] CRAN (R 4.1.0)
##  ellipsis                            0.3.0    2019-09-20 [2] CRAN (R 4.1.0)
##  ensembldb                           2.11.4   2020-04-17 [1] Bioconductor  
##  evaluate                            0.14     2019-05-28 [2] CRAN (R 4.1.0)
##  fansi                               0.4.1    2020-01-08 [2] CRAN (R 4.1.0)
##  farver                              2.0.3    2020-01-16 [1] CRAN (R 4.1.0)
##  foreach                           * 1.5.0    2020-03-30 [1] CRAN (R 4.1.0)
##  foreign                             0.8-78   2020-04-13 [3] CRAN (R 4.1.0)
##  Formula                             1.2-3    2018-05-03 [1] CRAN (R 4.1.0)
##  fs                                  1.4.1    2020-04-04 [2] CRAN (R 4.1.0)
##  generics                            0.0.2    2018-11-29 [1] CRAN (R 4.1.0)
##  GenomeInfoDb                      * 1.23.17  2020-04-13 [1] Bioconductor  
##  GenomeInfoDbData                    1.2.3    2020-04-25 [1] Bioconductor  
##  GenomicAlignments                   1.23.2   2020-03-24 [1] Bioconductor  
##  GenomicFeatures                   * 1.39.7   2020-03-19 [1] Bioconductor  
##  GenomicFiles                        1.23.2   2020-03-23 [1] Bioconductor  
##  GenomicRanges                     * 1.39.3   2020-03-24 [1] Bioconductor  
##  GGally                              1.5.0    2020-03-25 [1] CRAN (R 4.1.0)
##  ggbio                               1.35.2   2020-04-09 [1] Bioconductor  
##  ggplot2                             3.3.0    2020-03-05 [1] CRAN (R 4.1.0)
##  glue                                1.4.0    2020-04-03 [2] CRAN (R 4.1.0)
##  graph                               1.65.3   2020-04-14 [1] Bioconductor  
##  gridExtra                           2.3      2017-09-09 [1] CRAN (R 4.1.0)
##  gtable                              0.3.0    2019-03-25 [1] CRAN (R 4.1.0)
##  highr                               0.8      2019-03-20 [2] CRAN (R 4.1.0)
##  Hmisc                               4.4-0    2020-03-23 [1] CRAN (R 4.1.0)
##  hms                                 0.5.3    2020-01-08 [1] CRAN (R 4.1.0)
##  htmlTable                           1.13.3   2019-12-04 [1] CRAN (R 4.1.0)
##  htmltools                           0.4.0    2019-10-04 [2] CRAN (R 4.1.0)
##  htmlwidgets                         1.5.1    2019-10-08 [2] CRAN (R 4.1.0)
##  httr                                1.4.1    2019-08-05 [2] CRAN (R 4.1.0)
##  IRanges                           * 2.21.8   2020-03-25 [1] Bioconductor  
##  iterators                         * 1.0.12   2019-07-26 [1] CRAN (R 4.1.0)
##  jpeg                                0.1-8.1  2019-10-24 [1] CRAN (R 4.1.0)
##  jsonlite                            1.6.1    2020-02-02 [2] CRAN (R 4.1.0)
##  knitcitations                     * 1.0.10   2019-09-15 [1] CRAN (R 4.1.0)
##  knitr                             * 1.28     2020-02-06 [2] CRAN (R 4.1.0)
##  labeling                            0.3      2014-08-23 [1] CRAN (R 4.1.0)
##  lattice                             0.20-41  2020-04-02 [3] CRAN (R 4.1.0)
##  latticeExtra                        0.6-29   2019-12-19 [1] CRAN (R 4.1.0)
##  lazyeval                            0.2.2    2019-03-15 [2] CRAN (R 4.1.0)
##  lifecycle                           0.2.0    2020-03-06 [2] CRAN (R 4.1.0)
##  limma                               3.43.8   2020-04-14 [1] Bioconductor  
##  locfit                            * 1.5-9.4  2020-03-25 [1] CRAN (R 4.1.0)
##  lubridate                           1.7.8    2020-04-06 [1] CRAN (R 4.1.0)
##  magrittr                            1.5      2014-11-22 [2] CRAN (R 4.1.0)
##  MASS                                7.3-51.5 2019-12-20 [3] CRAN (R 4.1.0)
##  Matrix                              1.2-18   2019-11-27 [3] CRAN (R 4.1.0)
##  matrixStats                         0.56.0   2020-03-13 [1] CRAN (R 4.1.0)
##  memoise                             1.1.0    2017-04-21 [2] CRAN (R 4.1.0)
##  munsell                             0.5.0    2018-06-12 [1] CRAN (R 4.1.0)
##  nnet                                7.3-13   2020-02-25 [3] CRAN (R 4.1.0)
##  openssl                             1.4.1    2019-07-18 [2] CRAN (R 4.1.0)
##  org.Hs.eg.db                      * 3.10.0   2020-04-25 [1] Bioconductor  
##  OrganismDbi                         1.29.1   2020-02-04 [1] Bioconductor  
##  pillar                              1.4.3    2019-12-20 [2] CRAN (R 4.1.0)
##  pkgconfig                           2.0.3    2019-09-22 [2] CRAN (R 4.1.0)
##  pkgdown                             1.5.1    2020-04-09 [1] CRAN (R 4.1.0)
##  plyr                                1.8.6    2020-03-03 [1] CRAN (R 4.1.0)
##  png                                 0.1-7    2013-12-03 [1] CRAN (R 4.1.0)
##  prettyunits                         1.1.1    2020-01-24 [2] CRAN (R 4.1.0)
##  progress                            1.2.2    2019-05-16 [1] CRAN (R 4.1.0)
##  ProtGenerics                        1.19.3   2019-12-25 [1] Bioconductor  
##  purrr                               0.3.4    2020-04-17 [2] CRAN (R 4.1.0)
##  qvalue                              2.19.0   2019-10-29 [1] Bioconductor  
##  R6                                  2.4.1    2019-11-12 [2] CRAN (R 4.1.0)
##  rappdirs                            0.3.1    2016-03-28 [1] CRAN (R 4.1.0)
##  RBGL                                1.63.1   2019-10-30 [1] Bioconductor  
##  RColorBrewer                        1.1-2    2014-12-07 [1] CRAN (R 4.1.0)
##  Rcpp                                1.0.4.6  2020-04-09 [2] CRAN (R 4.1.0)
##  RCurl                               1.98-1.2 2020-04-18 [1] CRAN (R 4.1.0)
##  RefManageR                          1.2.12   2019-04-03 [1] CRAN (R 4.1.0)
##  reshape                             0.8.8    2018-10-23 [1] CRAN (R 4.1.0)
##  reshape2                            1.4.4    2020-04-09 [1] CRAN (R 4.1.0)
##  rlang                               0.4.5    2020-03-01 [2] CRAN (R 4.1.0)
##  rmarkdown                           2.1      2020-01-20 [1] CRAN (R 4.1.0)
##  rngtools                            1.5      2020-01-23 [1] CRAN (R 4.1.0)
##  rpart                               4.1-15   2019-04-12 [3] CRAN (R 4.1.0)
##  rprojroot                           1.3-2    2018-01-03 [2] CRAN (R 4.1.0)
##  Rsamtools                           2.3.7    2020-03-18 [1] Bioconductor  
##  RSQLite                             2.2.0    2020-01-07 [1] CRAN (R 4.1.0)
##  rstudioapi                          0.11     2020-02-07 [2] CRAN (R 4.1.0)
##  rtracklayer                         1.47.0   2019-10-29 [1] Bioconductor  
##  S4Vectors                         * 0.25.15  2020-04-04 [1] Bioconductor  
##  scales                              1.1.0    2019-11-18 [1] CRAN (R 4.1.0)
##  sessioninfo                       * 1.1.1    2018-11-05 [2] CRAN (R 4.1.0)
##  stringi                             1.4.6    2020-02-17 [2] CRAN (R 4.1.0)
##  stringr                             1.4.0    2019-02-10 [2] CRAN (R 4.1.0)
##  SummarizedExperiment                1.17.5   2020-03-27 [1] Bioconductor  
##  survival                            3.1-12   2020-04-10 [3] CRAN (R 4.1.0)
##  tibble                              3.0.1    2020-04-20 [2] CRAN (R 4.1.0)
##  tidyselect                          1.0.0    2020-01-27 [1] CRAN (R 4.1.0)
##  TxDb.Hsapiens.UCSC.hg19.knownGene * 3.2.2    2020-04-25 [1] Bioconductor  
##  VariantAnnotation                   1.33.5   2020-04-23 [1] Bioconductor  
##  vctrs                               0.2.4    2020-03-10 [2] CRAN (R 4.1.0)
##  withr                               2.2.0    2020-04-20 [2] CRAN (R 4.1.0)
##  xfun                                0.13     2020-04-13 [2] CRAN (R 4.1.0)
##  XML                                 3.99-0.3 2020-01-20 [1] CRAN (R 4.1.0)
##  xml2                                1.3.2    2020-04-23 [2] CRAN (R 4.1.0)
##  XVector                             0.27.2   2020-03-24 [1] Bioconductor  
##  yaml                                2.2.1    2020-02-01 [2] CRAN (R 4.1.0)
##  zlibbioc                            1.33.1   2020-01-24 [1] Bioconductor  
## 
## [1] /usr/local/lib/R/host-site-library
## [2] /usr/local/lib/R/site-library
## [3] /usr/local/lib/R/library</code></pre>
</div>
<div id="bibliography" class="section level1">
<h1 class="hasAnchor">
<a href="#bibliography" class="anchor"></a>Bibliography</h1>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.11/BiocStyle">BiocStyle</a></em> (<a href="https://github.com/Bioconductor/BiocStyle">Oleś, Morgan, and Huber, 2020</a>) with <em><a href="https://CRAN.R-project.org/package=knitr">knitr</a></em> (<a href="http://www.crcpress.com/product/isbn/9781466561595">Xie, 2014</a>) and <em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em> (<a href="https://github.com/rstudio/rmarkdown">Allaire, Xie, McPherson, Luraschi, et al., 2020</a>) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=knitcitations">knitcitations</a></em> (<a href="https://CRAN.R-project.org/package=knitcitations">Boettiger, 2019</a>).</p>
<p>[1] J. Allaire, Y. Xie, J. McPherson, J. Luraschi, et al. <em>rmarkdown: Dynamic Documents for R</em>. R package version 2.1. 2020. &lt;URL: <a href="https://github.com/rstudio/rmarkdown" class="uri">https://github.com/rstudio/rmarkdown</a>&gt;.</p>
<p>[2] S. Arora, M. Morgan, M. Carlson, and H. Pagès. <em>GenomeInfoDb: Utilities for manipulating chromosome and other ‘seqname’ identifiers</em>. 2017. DOI: 10.18129/B9.bioc.GenomeInfoDb.</p>
<p>[3] C. Boettiger. <em>knitcitations: Citations for ‘Knitr’ Markdown Files</em>. R package version 1.0.10. 2019. &lt;URL: <a href="https://CRAN.R-project.org/package=knitcitations" class="uri">https://CRAN.R-project.org/package=knitcitations</a>&gt;.</p>
<p>[4] BrainSpan. “Atlas of the Developing Human Brain [Internet]. Funded by ARRA Awards 1RC2MH089921-01, 1RC2MH090047-01, and 1RC2MH089929-01.” 2011. &lt;URL: <a href="http://www.brainspan.org/" class="uri">http://www.brainspan.org/</a>&gt;.</p>
<p>[5] M. Carlson and B. P. Maintainer. <em>TxDb.Hsapiens.UCSC.hg19.knownGene: Annotation package for TxDb object(s)</em>. R package version 3.2.2. 2015.</p>
<p>[6] L. Collado-Torres, A. E. Jaffe, and J. T. Leek. <em>derfinderPlot: Plotting functions for derfinder</em>. <a href="https://github.com/leekgroup/derfinderPlot" class="uri">https://github.com/leekgroup/derfinderPlot</a> - R package version 1.21.5. 2017. DOI: 10.18129/B9.bioc.derfinderPlot. &lt;URL: <a href="http://www.bioconductor.org/packages/derfinderPlot" class="uri">http://www.bioconductor.org/packages/derfinderPlot</a>&gt;.</p>
<p>[7] L. Collado-Torres, A. Jaffe, and J. Leek. <em>derfinderData: Processed BigWigs from BrainSpan for examples</em>. R package version 2.5.3. 2020. &lt;URL: <a href="https://github.com/leekgroup/derfinderData" class="uri">https://github.com/leekgroup/derfinderData</a>&gt;.</p>
<p>[8] L. Collado-Torres, A. Nellore, A. C. Frazee, C. Wilks, et al. “Flexible expressed region analysis for RNA-seq with derfinder”. In: <em>Nucl. Acids Res.</em> (2017). DOI: 10.1093/nar/gkw852. &lt;URL: <a href="http://nar.oxfordjournals.org/content/early/2016/09/29/nar.gkw852" class="uri">http://nar.oxfordjournals.org/content/early/2016/09/29/nar.gkw852</a>&gt;.</p>
<p>[9] G. Csárdi, R. core, H. Wickham, W. Chang, et al. <em>sessioninfo: R Session Information</em>. R package version 1.1.1. 2018. &lt;URL: <a href="https://CRAN.R-project.org/package=sessioninfo" class="uri">https://CRAN.R-project.org/package=sessioninfo</a>&gt;.</p>
<p>[10] A. E. Jaffe, P. Murakami, H. Lee, J. T. Leek, et al. “Bump hunting to identify differentially methylated regions in epigenetic epidemiology studies”. In: <em>International journal of epidemiology</em> 41.1 (2012), pp. 200-209. DOI: 10.1093/ije/dyr238.</p>
<p>[11] A. E. Jaffe, P. Murakami, H. Lee, J. T. Leek, et al. “Bump hunting to identify differentially methylated regions in epigenetic epidemiology studies”. In: <em>International Journal of Epidemiology</em> (2012).</p>
<p>[12] M. Lawrence, W. Huber, H. Pagès, P. Aboyoun, et al. “Software for Computing and Annotating Genomic Ranges”. In: <em>PLoS Computational Biology</em> 9 (8 2013). DOI: 10.1371/journal.pcbi.1003118. &lt;URL: <a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118" class="uri">http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118</a>}.&gt;</p>
<p>[13] M. Lawrence, W. Huber, H. Pagès, P. Aboyoun, et al. “Software for Computing and Annotating Genomic Ranges”. In: <em>PLoS Computational Biology</em> 9 (8 2013). DOI: 10.1371/journal.pcbi.1003118. &lt;URL: <a href="http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118" class="uri">http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003118</a>}.&gt;</p>
<p>[14] E. Neuwirth. <em>RColorBrewer: ColorBrewer Palettes</em>. R package version 1.1-2. 2014. &lt;URL: <a href="https://CRAN.R-project.org/package=RColorBrewer" class="uri">https://CRAN.R-project.org/package=RColorBrewer</a>&gt;.</p>
<p>[15] A. Oleś, M. Morgan, and W. Huber. <em>BiocStyle: Standard styles for vignettes and other Bioconductor documents</em>. R package version 2.15.8. 2020. &lt;URL: <a href="https://github.com/Bioconductor/BiocStyle" class="uri">https://github.com/Bioconductor/BiocStyle</a>&gt;.</p>
<p>[16] R Core Team. <em>R: A Language and Environment for Statistical Computing</em>. R Foundation for Statistical Computing. Vienna, Austria, 2020. &lt;URL: <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>&gt;.</p>
<p>[17] H. Wickham. <em>ggplot2: Elegant Graphics for Data Analysis</em>. Springer-Verlag New York, 2016. ISBN: 978-3-319-24277-4. &lt;URL: <a href="https://ggplot2.tidyverse.org" class="uri">https://ggplot2.tidyverse.org</a>&gt;.</p>
<p>[18] H. Wickham. “Reshaping Data with the reshape Package”. In: <em>Journal of Statistical Software</em> 21.12 (2007), pp. 1-20. &lt;URL: <a href="http://www.jstatsoft.org/v21/i12/" class="uri">http://www.jstatsoft.org/v21/i12/</a>&gt;.</p>
<p>[19] H. Wickham. “testthat: Get Started with Testing”. In: <em>The R Journal</em> 3 (2011), pp. 5-10. &lt;URL: <a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="uri">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>&gt;.</p>
<p>[20] H. Wickham. “The Split-Apply-Combine Strategy for Data Analysis”. In: <em>Journal of Statistical Software</em> 40.1 (2011), pp. 1-29. &lt;URL: <a href="http://www.jstatsoft.org/v40/i01/" class="uri">http://www.jstatsoft.org/v40/i01/</a>&gt;.</p>
<p>[21] H. Wickham and D. Seidel. <em>scales: Scale Functions for Visualization</em>. R package version 1.1.0. 2019. &lt;URL: <a href="https://CRAN.R-project.org/package=scales" class="uri">https://CRAN.R-project.org/package=scales</a>&gt;.</p>
<p>[22] Y. Xie. “knitr: A Comprehensive Tool for Reproducible Research in R”. In: <em>Implementing Reproducible Computational Research</em>. Ed. by V. Stodden, F. Leisch and R. D. Peng. ISBN 978-1466561595. Chapman and Hall/CRC, 2014. &lt;URL: <a href="http://www.crcpress.com/product/isbn/9781466561595" class="uri">http://www.crcpress.com/product/isbn/9781466561595</a>&gt;.</p>
<p>[23] T. Yin, D. Cook, and M. Lawrence. “ggbio: an R package for extending the grammar of graphics for genomic data”. In: <em>Genome Biology</em> 13.8 (2012), p. R77.</p>
<p>[24] T. Yin, M. Lawrence, and D. Cook. <em>biovizBase: Basic graphic utilities for visualization of genomic data.</em> R package version 1.35.1. 2019.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Leonardo Collado-Torres, Andrew E. Jaffe, Jeffrey T. Leek.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
